In [3]:
from sklearn.cross_validation import train_test_split
from sklearn.metrics import classification_report
from sklearn.externals import joblib
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix
from sklearn import datasets
from nolearn.dbn import DBN
from skimage import io
from skimage import data, segmentation, filters, color, img_as_float, img_as_ubyte, exposure, feature, measure, morphology
from skimage.color import rgb2gray
from skimage.morphology import square
from skimage.feature import hog
import cv2
import numpy as np
import cv2
from time import time
from glob import glob
import os

gnumpy: failed to import cudamat. Using npmat instead. No GPU will be used.


In [ ]:
### Plan2 ###

In [217]:
features = joblib.load("./mldata/features_1200.mat")
labels = joblib.load("./mldata/lables_1200.mat")

features = np.array(features, 'int16')
labels = np.array(labels, 'int')

In [218]:
t0 = time()
def scale(X, eps = 0.001):
    # scale the data points s.t the columns of the feature space
    # (i.e the predictors) are within the range [0, 1]
    return (X - np.min(X, axis = 0)) / (np.max(X, axis = 0) + eps)

features = features.astype("float32")
features = scale(features)

print "escape time : ", round(time()-t0, 3), "s"

escape time :  0.091 s


In [219]:
# scale the data to the range [0, 1] and then construct the training
# and testing splits
(trainX, testX, trainY, testY) = train_test_split(features, labels, test_size = 0.1)
print "the shape of training set %s rows, %s columns" %(trainX.shape[0], trainX.shape[1])
print "the shape of test set %s rows, %s columns" %(testX.shape[0], testX.shape[1])
print "the range of training set : %s ~ %s" %(trainX.min(),trainX.max())
print "the range of test set : %s ~ %s" %(testX.min(),testX.max())

the shape of training set 5400 rows, 784 columns
the shape of test set 600 rows, 784 columns
the range of training set : 0.0 ~ 0.999996
the range of test set : 0.0 ~ 0.999996


In [220]:
# train the Deep Belief Network with 784 input units (the flattened,
# 28x28 grayscale image), 500 hidden units, 9 output units (one for
# each possible output classification, which are the digits 0-8)
t0 = time()
dbn = DBN(
    [trainX.shape[1], 500, 9],
    learn_rates = 0.009,
    learn_rate_decays = 0.9,
    epochs = 10,
    verbose = 1)
dbn.fit(trainX, trainY)
print "escape time : ", round(time()-t0, 3), "s"

[DBN] fitting X.shape=(5400, 784)
[DBN] layers [784, 500, 9]
[DBN] Fine-tune...

100%



Epoch 1:

100%



  loss 0.463643961276
  err  0.115513392857
  (0:00:01)
Epoch 2:

100%



  loss 0.29921507587
  err  0.073474702381
  (0:00:01)
Epoch 3:

100%



  loss 0.277418208442
  err  0.0719866071429
  (0:00:01)
Epoch 4:

100%



  loss 0.251923746917
  err  0.0641741071429
  (0:00:01)
Epoch 5:

100%



  loss 0.24042045875
  err  0.0651041666667
  (0:00:01)
Epoch 6:

100%



  loss 0.240729313699
  err  0.0652901785714
  (0:00:01)
Epoch 7:

100%



  loss 0.214639534109
  err  0.0574776785714
  (0:00:01)
Epoch 8:

100%



  loss 0.235282459518
  err  0.0638020833333
  (0:00:01)
Epoch 9:

100%



  loss 0.20613166548
  err  0.0580357142857
  (0:00:01)
Epoch 10:
  loss 0.208449641597
  err  0.0556175595238
  (0:00:01)
escape time :  12.585 s


In [221]:
# compute the predictions for the test data and show a classification report
preds = dbn.predict(testX)
print "accuracy score : %s" %(accuracy_score(testY, preds))
print "classification report : "
print classification_report(testY, preds)
print "confusion matrix : "
print confusion_matrix(testY, preds)

accuracy score : 0.95
classification report : 
             precision    recall  f1-score   support

          0       0.96      0.93      0.94        70
          1       0.93      0.96      0.94        93
          2       0.86      0.93      0.89        55
          3       0.98      0.95      0.96        58
          4       0.98      0.98      0.98        64
          5       0.98      0.94      0.96        62
          6       0.90      0.84      0.87        45
          7       1.00      0.99      0.99       100
          8       0.91      0.98      0.95        53

avg / total       0.95      0.95      0.95       600

confusion matrix : 
[[65  0  3  0  0  0  1  0  1]
 [ 2 89  1  0  0  0  0  0  1]
 [ 0  1 51  1  0  0  1  0  1]
 [ 0  1  2 55  0  0  0  0  0]
 [ 0  0  0  0 63  0  0  0  1]
 [ 0  2  1  0  0 58  1  0  0]
 [ 1  3  1  0  0  1 38  0  1]
 [ 0  0  0  0  0  0  1 99  0]
 [ 0  0  0  0  1  0  0  0 52]]


In [1]:
# MNIST TEST

In [4]:
dataset = datasets.fetch_mldata("MNIST Original")

# scale the data to the range [0, 1] and then construct the training
# and testing splits
(trainX, testX, trainY, testY) = train_test_split(dataset.data / 255.0, dataset.target.astype("int0"), test_size = 0.33)

In [16]:
# train the Deep Belief Network with 784 input units (the flattened,
# 28x28 grayscale image), 300 hidden units, 10 output units (one for
# each possible output classification, which are the digits 1-10)
dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.01,
    learn_rate_decays = 0.9,
    epochs = 1000,
    verbose = 1)
dbn.fit(trainX, trainY)

# compute the predictions for the test data and show a classification report
preds = dbn.predict(testX)
print "accuracy score : %s" %(accuracy_score(testY, preds))
print "classification report : "
print classification_report(testY, preds)
print "confusion matrix : "
print confusion_matrix(testY, preds)

[DBN] fitting X.shape=(46900, 784)
[DBN] layers [784, 300, 10]
[DBN] Fine-tune...

100%



Epoch 1:

100%



  loss 0.435460707254
  err  0.121285860656
  (0:00:07)
Epoch 2:

100%



  loss 0.222520561655
  err  0.0626494193989
  (0:00:06)
Epoch 3:

100%



  loss 0.172146648504
  err  0.0493510928962
  (0:00:06)
Epoch 4:

100%



  loss 0.141563853361
  err  0.0399163251366
  (0:00:06)
Epoch 5:

100%



  loss 0.122119528071
  err  0.0332137978142
  (0:00:06)
Epoch 6:

100%



  loss 0.107311866964
  err  0.0295210040984
  (0:00:06)
Epoch 7:

100%



  loss 0.103555048722
  err  0.028112192623
  (0:00:06)
Epoch 8:

100%



  loss 0.0913765825397
  err  0.0241632513661
  (0:00:06)
Epoch 9:

100%



  loss 0.0862452716508
  err  0.0232453893443
  (0:00:07)
Epoch 10:

100%



  loss 0.0784475447819
  err  0.0198941256831
  (0:00:06)
Epoch 11:

100%



  loss 0.0759711375631
  err  0.0191256830601
  (0:00:06)
Epoch 12:

100%



  loss 0.0740048806705
  err  0.0200008538251
  (0:00:07)
Epoch 13:

100%



  loss 0.0682661257814
  err  0.0176741803279
  (0:00:06)
Epoch 14:

100%



  loss 0.0682453495747
  err  0.0175461065574
  (0:00:07)
Epoch 15:

100%



  loss 0.0668429842885
  err  0.0175887978142
  (0:00:06)
Epoch 16:

100%



  loss 0.0621177279034
  err  0.0152194330601
  (0:00:07)
Epoch 17:

100%



  loss 0.0612762555432
  err  0.015881147541
  (0:00:07)
Epoch 18:

100%



  loss 0.062244954762
  err  0.0155823087432
  (0:00:06)
Epoch 19:

100%



  loss 0.0591402238711
  err  0.0145150273224
  (0:00:06)
Epoch 20:

100%



  loss 0.0599976112765
  err  0.0147284836066
  (0:00:06)
Epoch 21:

100%



  loss 0.0565000520752
  err  0.0140454234973
  (0:00:06)
Epoch 22:

100%



  loss 0.0570156935048
  err  0.0134904371585
  (0:00:06)
Epoch 23:

100%



  loss 0.0558370767464
  err  0.0137038934426
  (0:00:06)
Epoch 24:

100%



  loss 0.0547825844292
  err  0.0133837090164
  (0:00:06)
Epoch 25:

100%



  loss 0.0577150893212
  err  0.0141521516393
  (0:00:06)
Epoch 26:

100%



  loss 0.0532263154317
  err  0.0122950819672
  (0:00:06)
Epoch 27:

100%



  loss 0.0534376011477
  err  0.0124018101093
  (0:00:06)
Epoch 28:

100%



  loss 0.0553420051943
  err  0.0135971653005
  (0:00:06)
Epoch 29:

100%



  loss 0.050837528402
  err  0.0119535519126
  (0:00:06)
Epoch 30:

100%



  loss 0.0526713378628
  err  0.011974897541
  (0:00:06)
Epoch 31:

100%



  loss 0.0529029441506
  err  0.0124658469945
  (0:00:06)
Epoch 32:

100%



  loss 0.0531268519091
  err  0.0129781420765
  (0:00:06)
Epoch 33:

100%



  loss 0.0507176623131
  err  0.0119535519126
  (0:00:06)
Epoch 34:

100%



  loss 0.0499105089722
  err  0.0120175887978
  (0:00:06)
Epoch 35:

100%



  loss 0.0524979259837
  err  0.012593920765
  (0:00:06)
Epoch 36:

100%



  loss 0.0536942167298
  err  0.0126793032787
  (0:00:06)
Epoch 37:

100%



  loss 0.0529735623693
  err  0.0135117827869
  (0:00:07)
Epoch 38:

100%



  loss 0.0515139730306
  err  0.0122737363388
  (0:00:06)
Epoch 39:

100%



  loss 0.0508582337557
  err  0.0113558743169
  (0:00:06)
Epoch 40:

100%



  loss 0.0512382912035
  err  0.011974897541
  (0:00:07)
Epoch 41:

100%



  loss 0.0522620541901
  err  0.0132556352459
  (0:00:07)
Epoch 42:

100%



  loss 0.049911510191
  err  0.0116974043716
  (0:00:07)
Epoch 43:

100%



  loss 0.0515477209444
  err  0.0122096994536
  (0:00:06)
Epoch 44:

100%



  loss 0.0508786643413
  err  0.0120602800546
  (0:00:06)
Epoch 45:

100%



  loss 0.0511269245622
  err  0.0120389344262
  (0:00:07)
Epoch 46:

100%



  loss 0.0491818494629
  err  0.0118041325137
  (0:00:07)
Epoch 47:

100%



  loss 0.0503594469218
  err  0.0116333674863
  (0:00:07)
Epoch 48:

100%



  loss 0.0516768718607
  err  0.0121243169399
  (0:00:07)
Epoch 49:

100%



  loss 0.0495825724809
  err  0.0114839480874
  (0:00:07)
Epoch 50:

100%



  loss 0.0489350383868
  err  0.0111210724044
  (0:00:07)
Epoch 51:

100%



  loss 0.0515095751174
  err  0.0122737363388
  (0:00:06)
Epoch 52:

100%



  loss 0.0523622185742
  err  0.012850068306
  (0:00:06)
Epoch 53:

100%



  loss 0.051790614211
  err  0.0125512295082
  (0:00:06)
Epoch 54:

100%



  loss 0.0487533225379
  err  0.0107795423497
  (0:00:07)
Epoch 55:

100%



  loss 0.0507027208411
  err  0.0118254781421
  (0:00:06)
Epoch 56:

100%



  loss 0.0500482111492
  err  0.0118681693989
  (0:00:06)
Epoch 57:

100%



  loss 0.0507887408393
  err  0.0121883538251
  (0:00:06)
Epoch 58:

100%



  loss 0.0515041196378
  err  0.0116760587432
  (0:00:06)
Epoch 59:

100%



  loss 0.0527089130925
  err  0.0126579576503
  (0:00:06)
Epoch 60:

100%



  loss 0.0494819635963
  err  0.011462602459
  (0:00:06)
Epoch 61:

100%



  loss 0.0505767093419
  err  0.0116974043716
  (0:00:06)
Epoch 62:

100%



  loss 0.0528903364247
  err  0.0125512295082
  (0:00:06)
Epoch 63:

100%



  loss 0.050791340048
  err  0.0119535519126
  (0:00:06)
Epoch 64:

100%



  loss 0.0513245650145
  err  0.011974897541
  (0:00:07)
Epoch 65:

100%



  loss 0.0519084002335
  err  0.012593920765
  (0:00:07)
Epoch 66:

100%



  loss 0.0525383493059
  err  0.0122950819672
  (0:00:07)
Epoch 67:

100%



  loss 0.0488930429864
  err  0.0115479849727
  (0:00:07)
Epoch 68:

100%



  loss 0.0481096729649
  err  0.0107795423497
  (0:00:06)
Epoch 69:

100%



  loss 0.04927954286
  err  0.0118681693989
  (0:00:06)
Epoch 70:

100%



  loss 0.0493161196138
  err  0.01171875
  (0:00:06)
Epoch 71:

100%



  loss 0.051179053669
  err  0.0124231557377
  (0:00:06)
Epoch 72:

100%



  loss 0.0502255971988
  err  0.0115693306011
  (0:00:06)
Epoch 73:

100%



  loss 0.0488801487795
  err  0.0113131830601
  (0:00:06)
Epoch 74:

100%



  loss 0.0507979828093
  err  0.0123804644809
  (0:00:06)
Epoch 75:

100%



  loss 0.0517147362781
  err  0.0124658469945
  (0:00:06)
Epoch 76:

100%



  loss 0.0492344016535
  err  0.0116120218579
  (0:00:06)
Epoch 77:

100%



  loss 0.0485727893554
  err  0.0117614412568
  (0:00:06)
Epoch 78:

100%



  loss 0.0502685980936
  err  0.0120602800546
  (0:00:06)
Epoch 79:

100%



  loss 0.051556781552
  err  0.0126366120219
  (0:00:06)
Epoch 80:

100%



  loss 0.0492101134185
  err  0.0110356898907
  (0:00:07)
Epoch 81:

100%



  loss 0.0488571881436
  err  0.0114199112022
  (0:00:06)
Epoch 82:

100%



  loss 0.0504408535723
  err  0.0119108606557
  (0:00:07)
Epoch 83:

100%



  loss 0.0480929198446
  err  0.0112918374317
  (0:00:07)
Epoch 84:

100%



  loss 0.0486106668056
  err  0.0111637636612
  (0:00:07)
Epoch 85:

100%



  loss 0.0504939691768
  err  0.0123591188525
  (0:00:07)
Epoch 86:

100%



  loss 0.0505683805852
  err  0.0116333674863
  (0:00:07)
Epoch 87:

100%



  loss 0.0495635733833
  err  0.0122523907104
  (0:00:07)
Epoch 88:

100%



  loss 0.0484810289437
  err  0.0112704918033
  (0:00:07)
Epoch 89:

100%



  loss 0.0527064129488
  err  0.0125725751366
  (0:00:06)
Epoch 90:

100%



  loss 0.0482768714994
  err  0.0111424180328
  (0:00:07)
Epoch 91:

100%



  loss 0.0501981932216
  err  0.0116974043716
  (0:00:06)
Epoch 92:

100%



  loss 0.0503788712686
  err  0.0119108606557
  (0:00:07)
Epoch 93:

100%



  loss 0.0508723249449
  err  0.0124231557377
  (0:00:07)
Epoch 94:

100%



  loss 0.0498046751612
  err  0.0115906762295
  (0:00:07)
Epoch 95:

100%



  loss 0.0498483787365
  err  0.0118468237705
  (0:00:07)
Epoch 96:

100%



  loss 0.050015798324
  err  0.0119535519126
  (0:00:07)
Epoch 97:

100%



  loss 0.0497866803729
  err  0.0109929986339
  (0:00:08)
Epoch 98:

100%



  loss 0.0514068255504
  err  0.0122096994536
  (0:00:07)
Epoch 99:

100%



  loss 0.0499495779122
  err  0.0113345286885
  (0:00:07)
Epoch 100:

100%



  loss 0.049502778937
  err  0.0119962431694
  (0:00:07)
Epoch 101:

100%



  loss 0.0499846006418
  err  0.0114839480874
  (0:00:07)
Epoch 102:

100%



  loss 0.0499399278715
  err  0.0118041325137
  (0:00:07)
Epoch 103:

100%



  loss 0.0511863092597
  err  0.0118895150273
  (0:00:07)
Epoch 104:

100%



  loss 0.0504025924229
  err  0.0119108606557
  (0:00:07)
Epoch 105:

100%



  loss 0.0502416169388
  err  0.0114839480874
  (0:00:07)
Epoch 106:

100%



  loss 0.0491164478148
  err  0.0113772199454
  (0:00:07)
Epoch 107:

100%



  loss 0.0501602214196
  err  0.0113131830601
  (0:00:07)
Epoch 108:

100%



  loss 0.0487159733365
  err  0.010950307377
  (0:00:07)
Epoch 109:

100%



  loss 0.0509663925161
  err  0.0118468237705
  (0:00:06)
Epoch 110:

100%



  loss 0.0483703659567
  err  0.0110570355191
  (0:00:06)
Epoch 111:

100%



  loss 0.0526046921115
  err  0.0118041325137
  (0:00:06)
Epoch 112:

100%



  loss 0.0509754922572
  err  0.0123591188525
  (0:00:07)
Epoch 113:

100%



  loss 0.0491409050555
  err  0.0109289617486
  (0:00:07)
Epoch 114:

100%



  loss 0.0499582823043
  err  0.0113985655738
  (0:00:06)
Epoch 115:

100%



  loss 0.0492061110814
  err  0.0115479849727
  (0:00:07)
Epoch 116:

100%



  loss 0.0508090062576
  err  0.0123591188525
  (0:00:07)
Epoch 117:

100%



  loss 0.0502316899575
  err  0.0115693306011
  (0:00:07)
Epoch 118:

100%



  loss 0.0491954980131
  err  0.011206454918
  (0:00:07)
Epoch 119:

100%



  loss 0.0472271021422
  err  0.010843579235
  (0:00:07)
Epoch 120:

100%



  loss 0.0505263135601
  err  0.0118254781421
  (0:00:07)
Epoch 121:

100%



  loss 0.051360843469
  err  0.0115906762295
  (0:00:07)
Epoch 122:

100%



  loss 0.0487443197027
  err  0.0115052937158
  (0:00:07)
Epoch 123:

100%



  loss 0.0504776877402
  err  0.0113131830601
  (0:00:06)
Epoch 124:

100%



  loss 0.0494188256778
  err  0.011206454918
  (0:00:07)
Epoch 125:

100%



  loss 0.0509756106573
  err  0.0118468237705
  (0:00:07)
Epoch 126:

100%



  loss 0.0504256732243
  err  0.0115693306011
  (0:00:07)
Epoch 127:

100%



  loss 0.0509951044958
  err  0.0118254781421
  (0:00:07)
Epoch 128:

100%



  loss 0.0496411328605
  err  0.011974897541
  (0:00:06)
Epoch 129:

100%



  loss 0.0501996274348
  err  0.0115906762295
  (0:00:07)
Epoch 130:

100%



  loss 0.0507578475298
  err  0.0116547131148
  (0:00:06)
Epoch 131:

100%



  loss 0.0493191975198
  err  0.0116974043716
  (0:00:06)
Epoch 132:

100%



  loss 0.0493468159945
  err  0.0111210724044
  (0:00:07)
Epoch 133:

100%



  loss 0.0528174305819
  err  0.0127219945355
  (0:00:08)
Epoch 134:

100%



  loss 0.0499359456748
  err  0.0116120218579
  (0:00:09)
Epoch 135:

100%



  loss 0.0502331449238
  err  0.0115693306011
  (0:00:07)
Epoch 136:

100%



  loss 0.0497161746916
  err  0.0115266393443
  (0:00:07)
Epoch 137:

100%



  loss 0.0507891478432
  err  0.0119322062842
  (0:00:07)
Epoch 138:

100%



  loss 0.0524549048446
  err  0.0121243169399
  (0:00:07)
Epoch 139:

100%



  loss 0.050556717646
  err  0.0119535519126
  (0:00:07)
Epoch 140:

100%



  loss 0.0501314070218
  err  0.0117614412568
  (0:00:07)
Epoch 141:

100%



  loss 0.0499729871822
  err  0.0118895150273
  (0:00:07)
Epoch 142:

100%



  loss 0.0524305838817
  err  0.012487192623
  (0:00:07)
Epoch 143:

100%



  loss 0.0508016371085
  err  0.0122096994536
  (0:00:07)
Epoch 144:

100%



  loss 0.0487736085604
  err  0.0114839480874
  (0:00:07)
Epoch 145:

100%



  loss 0.0492291094842
  err  0.0112278005464
  (0:00:06)
Epoch 146:

100%



  loss 0.0510549071471
  err  0.0120816256831
  (0:00:07)
Epoch 147:

100%



  loss 0.0494225744446
  err  0.0114412568306
  (0:00:07)
Epoch 148:

100%



  loss 0.0481642678557
  err  0.0109716530055
  (0:00:07)
Epoch 149:

100%



  loss 0.0510784297548
  err  0.0122950819672
  (0:00:07)
Epoch 150:

100%



  loss 0.0509285974886
  err  0.0122096994536
  (0:00:07)
Epoch 151:

100%



  loss 0.049324102986
  err  0.01171875
  (0:00:06)
Epoch 152:

100%



  loss 0.0490994424256
  err  0.0114412568306
  (0:00:07)
Epoch 153:

100%



  loss 0.0480386513639
  err  0.0109929986339
  (0:00:07)
Epoch 154:

100%



  loss 0.0508999413533
  err  0.0120175887978
  (0:00:07)
Epoch 155:

100%



  loss 0.0503037777364
  err  0.0114412568306
  (0:00:07)
Epoch 156:

100%



  loss 0.0506360212041
  err  0.0113772199454
  (0:00:07)
Epoch 157:

100%



  loss 0.0513531301626
  err  0.0120602800546
  (0:00:07)
Epoch 158:

100%



  loss 0.049666257126
  err  0.0113558743169
  (0:00:07)
Epoch 159:

100%



  loss 0.051919741014
  err  0.0122950819672
  (0:00:07)
Epoch 160:

100%



  loss 0.0506863117694
  err  0.0117400956284
  (0:00:07)
Epoch 161:

100%



  loss 0.0510567859575
  err  0.0115052937158
  (0:00:07)
Epoch 162:

100%



  loss 0.0501529934989
  err  0.0118041325137
  (0:00:07)
Epoch 163:

100%



  loss 0.0495464904806
  err  0.0116120218579
  (0:00:07)
Epoch 164:

100%



  loss 0.0506851262034
  err  0.0118041325137
  (0:00:07)
Epoch 165:

100%



  loss 0.0497131138713
  err  0.0115693306011
  (0:00:07)
Epoch 166:

100%



  loss 0.0508165735728
  err  0.0122737363388
  (0:00:07)
Epoch 167:

100%



  loss 0.0504663349657
  err  0.0121029713115
  (0:00:07)
Epoch 168:

100%



  loss 0.0496776331524
  err  0.0109076161202
  (0:00:07)
Epoch 169:

100%



  loss 0.0502333519923
  err  0.0117400956284
  (0:00:07)
Epoch 170:

100%



  loss 0.0501132326314
  err  0.0121456625683
  (0:00:07)
Epoch 171:

100%



  loss 0.0484849282679
  err  0.0108862704918
  (0:00:07)
Epoch 172:

100%



  loss 0.0498018857054
  err  0.0113345286885
  (0:00:07)
Epoch 173:

100%



  loss 0.0505341369445
  err  0.0118895150273
  (0:00:07)
Epoch 174:

100%



  loss 0.0495719978016
  err  0.0115906762295
  (0:00:07)
Epoch 175:

100%



  loss 0.0514150606036
  err  0.01171875
  (0:00:07)
Epoch 176:

100%



  loss 0.0498682223648
  err  0.0117400956284
  (0:00:07)
Epoch 177:

100%



  loss 0.0514163339645
  err  0.0118895150273
  (0:00:07)
Epoch 178:

100%



  loss 0.0512010044255
  err  0.0124445013661
  (0:00:07)
Epoch 179:

100%



  loss 0.0486098712748
  err  0.0111210724044
  (0:00:07)
Epoch 180:

100%



  loss 0.0496798690601
  err  0.0113131830601
  (0:00:07)
Epoch 181:

100%



  loss 0.0503540183316
  err  0.0115266393443
  (0:00:07)
Epoch 182:

100%



  loss 0.0524816799252
  err  0.012487192623
  (0:00:08)
Epoch 183:

100%



  loss 0.051560939282
  err  0.0120602800546
  (0:00:07)
Epoch 184:

100%



  loss 0.0500074561019
  err  0.0122737363388
  (0:00:08)
Epoch 185:

100%



  loss 0.0498597675443
  err  0.0118254781421
  (0:00:09)
Epoch 186:

100%



  loss 0.0489506664008
  err  0.0118041325137
  (0:00:14)
Epoch 187:

100%



  loss 0.0499758651827
  err  0.0114839480874
  (0:00:08)
Epoch 188:

100%



  loss 0.0484744500352
  err  0.0117400956284
  (0:00:08)
Epoch 189:

100%



  loss 0.050884062563
  err  0.01171875
  (0:00:08)
Epoch 190:

100%



  loss 0.0500720708861
  err  0.0116120218579
  (0:00:07)
Epoch 191:

100%



  loss 0.0484472245908
  err  0.0110143442623
  (0:00:07)
Epoch 192:

100%



  loss 0.050796996949
  err  0.0123804644809
  (0:00:07)
Epoch 193:

100%



  loss 0.0490771630153
  err  0.0121456625683
  (0:00:07)
Epoch 194:

100%



  loss 0.0495293110169
  err  0.0113772199454
  (0:00:08)
Epoch 195:

100%



  loss 0.0487713256649
  err  0.0116974043716
  (0:00:07)
Epoch 196:

100%



  loss 0.050258159243
  err  0.0122950819672
  (0:00:07)
Epoch 197:

100%



  loss 0.0520754620965
  err  0.0121883538251
  (0:00:08)
Epoch 198:

100%



  loss 0.0507686280116
  err  0.0122737363388
  (0:00:08)
Epoch 199:

100%



  loss 0.0497930503611
  err  0.0118041325137
  (0:00:11)
Epoch 200:

100%



  loss 0.0498966457757
  err  0.0116547131148
  (0:00:08)
Epoch 201:

100%



  loss 0.051889432112
  err  0.0123804644809
  (0:00:07)
Epoch 202:

100%



  loss 0.0496989346906
  err  0.0118681693989
  (0:00:07)
Epoch 203:

100%



  loss 0.0506168593795
  err  0.01171875
  (0:00:07)
Epoch 204:

100%



  loss 0.0493622778564
  err  0.0111851092896
  (0:00:07)
Epoch 205:

100%



  loss 0.0499382914608
  err  0.0120175887978
  (0:00:07)
Epoch 206:

100%



  loss 0.0487163555408
  err  0.0107368510929
  (0:00:07)
Epoch 207:

100%



  loss 0.0490731110636
  err  0.011462602459
  (0:00:07)
Epoch 208:

100%



  loss 0.0495111989392
  err  0.0121456625683
  (0:00:07)
Epoch 209:

100%



  loss 0.0501960698207
  err  0.0120816256831
  (0:00:07)
Epoch 210:

100%



  loss 0.0510030075465
  err  0.012231045082
  (0:00:07)
Epoch 211:

100%



  loss 0.0502827696584
  err  0.0113131830601
  (0:00:08)
Epoch 212:

100%



  loss 0.0485733880908
  err  0.0110356898907
  (0:00:08)
Epoch 213:

100%



  loss 0.0490130639299
  err  0.0116333674863
  (0:00:07)
Epoch 214:

100%



  loss 0.0523703994816
  err  0.012593920765
  (0:00:08)
Epoch 215:

100%



  loss 0.0514407985299
  err  0.0123804644809
  (0:00:08)
Epoch 216:

100%



  loss 0.0486407127445
  err  0.0116547131148
  (0:00:07)
Epoch 217:

100%



  loss 0.0517211883873
  err  0.012231045082
  (0:00:08)
Epoch 218:

100%



  loss 0.0479020851071
  err  0.0112278005464
  (0:00:07)
Epoch 219:

100%



  loss 0.0503818490339
  err  0.0118254781421
  (0:00:07)
Epoch 220:

100%



  loss 0.0508013630465
  err  0.0119962431694
  (0:00:07)
Epoch 221:

100%



  loss 0.0494071773484
  err  0.0109289617486
  (0:00:07)
Epoch 222:

100%



  loss 0.0524357710647
  err  0.0123164275956
  (0:00:07)
Epoch 223:

100%



  loss 0.0492023112611
  err  0.0114412568306
  (0:00:07)
Epoch 224:

100%



  loss 0.0500198362581
  err  0.0116120218579
  (0:00:07)
Epoch 225:

100%



  loss 0.0499526950699
  err  0.0115906762295
  (0:00:07)
Epoch 226:

100%



  loss 0.0485320723555
  err  0.0110783811475
  (0:00:07)
Epoch 227:

100%



  loss 0.0494490827439
  err  0.0110143442623
  (0:00:07)
Epoch 228:

100%



  loss 0.0483264338099
  err  0.0115906762295
  (0:00:08)
Epoch 229:

100%



  loss 0.0515148152529
  err  0.0121456625683
  (0:00:08)
Epoch 230:

100%



  loss 0.0512057956145
  err  0.0120816256831
  (0:00:07)
Epoch 231:

100%



  loss 0.0499141030001
  err  0.0115693306011
  (0:00:07)
Epoch 232:

100%



  loss 0.0506959506998
  err  0.0116760587432
  (0:00:08)
Epoch 233:

100%



  loss 0.0509134919498
  err  0.0119535519126
  (0:00:07)
Epoch 234:

100%



  loss 0.048832004248
  err  0.0114412568306
  (0:00:08)
Epoch 235:

100%



  loss 0.0492413842101
  err  0.0112704918033
  (0:00:07)
Epoch 236:

100%



  loss 0.0505375803793
  err  0.011462602459
  (0:00:07)
Epoch 237:

100%



  loss 0.0532725324614
  err  0.0127219945355
  (0:00:09)
Epoch 238:

100%



  loss 0.04857445045
  err  0.0110570355191
  (0:00:08)
Epoch 239:

100%



  loss 0.047955694574
  err  0.0106514685792
  (0:00:08)
Epoch 240:

100%



  loss 0.0497003979447
  err  0.0116333674863
  (0:00:08)
Epoch 241:

100%



  loss 0.0492743251895
  err  0.0112704918033
  (0:00:07)
Epoch 242:

100%



  loss 0.0503058047104
  err  0.0116974043716
  (0:00:07)
Epoch 243:

100%



  loss 0.0501639536885
  err  0.0120602800546
  (0:00:07)
Epoch 244:

100%



  loss 0.0506415219546
  err  0.011462602459
  (0:00:07)
Epoch 245:

100%



  loss 0.049750947036
  err  0.0113772199454
  (0:00:07)
Epoch 246:

100%



  loss 0.049398807095
  err  0.0115266393443
  (0:00:07)
Epoch 247:

100%



  loss 0.0504627513783
  err  0.01171875
  (0:00:08)
Epoch 248:

100%



  loss 0.0480672939988
  err  0.0109716530055
  (0:00:07)
Epoch 249:

100%



  loss 0.0491115114158
  err  0.0123591188525
  (0:00:07)
Epoch 250:

100%



  loss 0.0519947745949
  err  0.0125298838798
  (0:00:07)
Epoch 251:

100%



  loss 0.0489623007795
  err  0.0115693306011
  (0:00:07)
Epoch 252:

100%



  loss 0.0506753178162
  err  0.0125298838798
  (0:00:07)
Epoch 253:

100%



  loss 0.0477509974391
  err  0.0103526297814
  (0:00:07)
Epoch 254:

100%



  loss 0.0499923775364
  err  0.01171875
  (0:00:07)
Epoch 255:

100%



  loss 0.0494499562354
  err  0.0116760587432
  (0:00:07)
Epoch 256:

100%



  loss 0.0498253315893
  err  0.0112704918033
  (0:00:07)
Epoch 257:

100%



  loss 0.0486054222381
  err  0.0115479849727
  (0:00:07)
Epoch 258:

100%



  loss 0.0511502098557
  err  0.0121456625683
  (0:00:07)
Epoch 259:

100%



  loss 0.0496056739691
  err  0.0114412568306
  (0:00:07)
Epoch 260:

100%



  loss 0.051989971487
  err  0.0121456625683
  (0:00:07)
Epoch 261:

100%



  loss 0.0499113416115
  err  0.0116547131148
  (0:00:07)
Epoch 262:

100%



  loss 0.0484211128625
  err  0.0115052937158
  (0:00:07)
Epoch 263:

100%



  loss 0.0506016829266
  err  0.0116333674863
  (0:00:07)
Epoch 264:

100%



  loss 0.0518718434859
  err  0.0120816256831
  (0:00:07)
Epoch 265:

100%



  loss 0.0525601512054
  err  0.0126793032787
  (0:00:07)
Epoch 266:

100%



  loss 0.0502177527788
  err  0.0119322062842
  (0:00:07)
Epoch 267:

100%



  loss 0.0507463470122
  err  0.0119108606557
  (0:00:07)
Epoch 268:

100%



  loss 0.0511011415335
  err  0.0124018101093
  (0:00:07)
Epoch 269:

100%



  loss 0.051296441847
  err  0.0117827868852
  (0:00:07)
Epoch 270:

100%



  loss 0.0490788285808
  err  0.0115479849727
  (0:00:07)
Epoch 271:

100%



  loss 0.051129538064
  err  0.0125085382514
  (0:00:07)
Epoch 272:

100%



  loss 0.0526003805651
  err  0.012593920765
  (0:00:07)
Epoch 273:

100%



  loss 0.0494846508201
  err  0.0114412568306
  (0:00:07)
Epoch 274:

100%



  loss 0.052696984837
  err  0.012231045082
  (0:00:07)
Epoch 275:

100%



  loss 0.0507760130268
  err  0.01171875
  (0:00:07)
Epoch 276:

100%



  loss 0.0536903757882
  err  0.0124231557377
  (0:00:07)
Epoch 277:

100%



  loss 0.0492575245406
  err  0.0114412568306
  (0:00:07)
Epoch 278:

100%



  loss 0.0521351535209
  err  0.0124018101093
  (0:00:07)
Epoch 279:

100%



  loss 0.0502439170333
  err  0.0121243169399
  (0:00:07)
Epoch 280:

100%



  loss 0.0494535245406
  err  0.01171875
  (0:00:07)
Epoch 281:

100%



  loss 0.0489470155113
  err  0.0116333674863
  (0:00:07)
Epoch 282:

100%



  loss 0.0501531602311
  err  0.0115693306011
  (0:00:07)
Epoch 283:

100%



  loss 0.0516170872009
  err  0.0121243169399
  (0:00:07)
Epoch 284:

100%



  loss 0.049269743585
  err  0.011462602459
  (0:00:07)
Epoch 285:

100%



  loss 0.0508077093317
  err  0.011974897541
  (0:00:07)
Epoch 286:

100%



  loss 0.0484791982911
  err  0.0120175887978
  (0:00:07)
Epoch 287:

100%



  loss 0.0498456783272
  err  0.0114199112022
  (0:00:07)
Epoch 288:

100%



  loss 0.051546293919
  err  0.012337773224
  (0:00:07)
Epoch 289:

100%



  loss 0.0503900251587
  err  0.0122737363388
  (0:00:07)
Epoch 290:

100%



  loss 0.0486616417982
  err  0.0116760587432
  (0:00:07)
Epoch 291:

100%



  loss 0.0485365100204
  err  0.0107368510929
  (0:00:07)
Epoch 292:

100%



  loss 0.0508744430377
  err  0.0118254781421
  (0:00:07)
Epoch 293:

100%



  loss 0.0504705181232
  err  0.0117400956284
  (0:00:07)
Epoch 294:

100%



  loss 0.0507223845607
  err  0.0119962431694
  (0:00:07)
Epoch 295:

100%



  loss 0.0497395926857
  err  0.0118254781421
  (0:00:07)
Epoch 296:

100%



  loss 0.0489204141224
  err  0.0109716530055
  (0:00:07)
Epoch 297:

100%



  loss 0.0496737950499
  err  0.011462602459
  (0:00:07)
Epoch 298:

100%



  loss 0.0507451051609
  err  0.01171875
  (0:00:07)
Epoch 299:

100%



  loss 0.0516190065219
  err  0.012593920765
  (0:00:07)
Epoch 300:

100%



  loss 0.0505587286882
  err  0.0116547131148
  (0:00:07)
Epoch 301:

100%



  loss 0.0506768477673
  err  0.0121456625683
  (0:00:07)
Epoch 302:

100%



  loss 0.0508807081719
  err  0.0123804644809
  (0:00:07)
Epoch 303:

100%



  loss 0.0499071399255
  err  0.0112278005464
  (0:00:07)
Epoch 304:

100%



  loss 0.0481445471002
  err  0.0112918374317
  (0:00:07)
Epoch 305:

100%



  loss 0.0509297787797
  err  0.0120389344262
  (0:00:07)
Epoch 306:

100%



  loss 0.0508494219479
  err  0.0117827868852
  (0:00:07)
Epoch 307:

100%



  loss 0.0500220229863
  err  0.0113985655738
  (0:00:07)
Epoch 308:

100%



  loss 0.050961653749
  err  0.0120175887978
  (0:00:07)
Epoch 309:

100%



  loss 0.0501931567951
  err  0.0124445013661
  (0:00:07)
Epoch 310:

100%



  loss 0.0512833137204
  err  0.0116760587432
  (0:00:07)
Epoch 311:

100%



  loss 0.0516443413636
  err  0.0123591188525
  (0:00:07)
Epoch 312:

100%



  loss 0.0523196896462
  err  0.0120816256831
  (0:00:07)
Epoch 313:

100%



  loss 0.0504238176849
  err  0.0119535519126
  (0:00:07)
Epoch 314:

100%



  loss 0.049805448993
  err  0.0116974043716
  (0:00:07)
Epoch 315:

100%



  loss 0.0514531745003
  err  0.0123804644809
  (0:00:07)
Epoch 316:

100%



  loss 0.0502288569003
  err  0.0117400956284
  (0:00:07)
Epoch 317:

100%



  loss 0.0493348478645
  err  0.0121029713115
  (0:00:07)
Epoch 318:

100%



  loss 0.0516280553729
  err  0.0127006489071
  (0:00:07)
Epoch 319:

100%



  loss 0.0497656506715
  err  0.0118468237705
  (0:00:07)
Epoch 320:

100%



  loss 0.0521773712816
  err  0.0122737363388
  (0:00:07)
Epoch 321:

100%



  loss 0.0483953345769
  err  0.0113772199454
  (0:00:07)
Epoch 322:

100%



  loss 0.0509135162056
  err  0.0121029713115
  (0:00:07)
Epoch 323:

100%



  loss 0.0506226576844
  err  0.0122950819672
  (0:00:07)
Epoch 324:

100%



  loss 0.051390163206
  err  0.012231045082
  (0:00:07)
Epoch 325:

100%



  loss 0.0508727383084
  err  0.0114412568306
  (0:00:07)
Epoch 326:

100%



  loss 0.0503910307498
  err  0.011974897541
  (0:00:07)
Epoch 327:

100%



  loss 0.0489122118722
  err  0.0111851092896
  (0:00:07)
Epoch 328:

100%



  loss 0.050651393124
  err  0.0115479849727
  (0:00:07)
Epoch 329:

100%



  loss 0.0494429865544
  err  0.0116333674863
  (0:00:07)
Epoch 330:

100%



  loss 0.0524348133725
  err  0.0123591188525
  (0:00:07)
Epoch 331:

100%



  loss 0.0505537139653
  err  0.0111637636612
  (0:00:07)
Epoch 332:

100%



  loss 0.0499254445064
  err  0.0112918374317
  (0:00:07)
Epoch 333:

100%



  loss 0.0499208709314
  err  0.0113345286885
  (0:00:07)
Epoch 334:

100%



  loss 0.0488756591088
  err  0.0107155054645
  (0:00:07)
Epoch 335:

100%



  loss 0.0479683054095
  err  0.0109716530055
  (0:00:07)
Epoch 336:

100%



  loss 0.050728998946
  err  0.0115906762295
  (0:00:07)
Epoch 337:

100%



  loss 0.0510064999737
  err  0.0119322062842
  (0:00:07)
Epoch 338:

100%



  loss 0.051085876675
  err  0.0118041325137
  (0:00:07)
Epoch 339:

100%



  loss 0.050164786002
  err  0.0121883538251
  (0:00:07)
Epoch 340:

100%



  loss 0.0504152302864
  err  0.0120602800546
  (0:00:07)
Epoch 341:

100%



  loss 0.0502447843287
  err  0.0116760587432
  (0:00:07)
Epoch 342:

100%



  loss 0.0497179747422
  err  0.0120602800546
  (0:00:07)
Epoch 343:

100%



  loss 0.0498885302143
  err  0.0116974043716
  (0:00:07)
Epoch 344:

100%



  loss 0.0501191260203
  err  0.0120389344262
  (0:00:07)
Epoch 345:

100%



  loss 0.0511252564452
  err  0.0117827868852
  (0:00:07)
Epoch 346:

100%



  loss 0.0493059469434
  err  0.0115906762295
  (0:00:07)
Epoch 347:

100%



  loss 0.0501769312207
  err  0.0120816256831
  (0:00:07)
Epoch 348:

100%



  loss 0.0491307397409
  err  0.0113772199454
  (0:00:07)
Epoch 349:

100%



  loss 0.0518051077347
  err  0.0122737363388
  (0:00:07)
Epoch 350:

100%



  loss 0.0503129398824
  err  0.0119535519126
  (0:00:07)
Epoch 351:

100%



  loss 0.0516837010423
  err  0.0119322062842
  (0:00:07)
Epoch 352:

100%



  loss 0.0512267150592
  err  0.0119535519126
  (0:00:07)
Epoch 353:

100%



  loss 0.0506162182492
  err  0.011974897541
  (0:00:07)
Epoch 354:

100%



  loss 0.0526611206587
  err  0.0124018101093
  (0:00:07)
Epoch 355:

100%



  loss 0.0500836158667
  err  0.010950307377
  (0:00:07)
Epoch 356:

100%



  loss 0.0506254300053
  err  0.0115266393443
  (0:00:07)
Epoch 357:

100%



  loss 0.051307229983
  err  0.0123804644809
  (0:00:07)
Epoch 358:

100%



  loss 0.0485731303766
  err  0.0115906762295
  (0:00:07)
Epoch 359:

100%



  loss 0.0511928648157
  err  0.0113772199454
  (0:00:07)
Epoch 360:

100%



  loss 0.0492339095227
  err  0.0112918374317
  (0:00:07)
Epoch 361:

100%



  loss 0.047597489704
  err  0.0108222336066
  (0:00:07)
Epoch 362:

100%



  loss 0.0510857033676
  err  0.011974897541
  (0:00:07)
Epoch 363:

100%



  loss 0.0498131474695
  err  0.0114412568306
  (0:00:07)
Epoch 364:

100%



  loss 0.0486089500365
  err  0.011099726776
  (0:00:07)
Epoch 365:

100%



  loss 0.049438437085
  err  0.0116120218579
  (0:00:07)
Epoch 366:

100%



  loss 0.0521541962685
  err  0.0119535519126
  (0:00:07)
Epoch 367:

100%



  loss 0.0514613784521
  err  0.0123164275956
  (0:00:07)
Epoch 368:

100%



  loss 0.0507776681296
  err  0.0117827868852
  (0:00:07)
Epoch 369:

100%



  loss 0.050779068922
  err  0.0124445013661
  (0:00:07)
Epoch 370:

100%



  loss 0.0500572536267
  err  0.011462602459
  (0:00:07)
Epoch 371:

100%



  loss 0.0497119690087
  err  0.0120816256831
  (0:00:07)
Epoch 372:

100%



  loss 0.0488072634977
  err  0.0112491461749
  (0:00:07)
Epoch 373:

100%



  loss 0.0497637483739
  err  0.0120816256831
  (0:00:07)
Epoch 374:

100%



  loss 0.0505072294112
  err  0.0120602800546
  (0:00:07)
Epoch 375:

100%



  loss 0.0495946167751
  err  0.0113985655738
  (0:00:07)
Epoch 376:

100%



  loss 0.050087231761
  err  0.0117400956284
  (0:00:07)
Epoch 377:

100%



  loss 0.0490173842342
  err  0.0112491461749
  (0:00:07)
Epoch 378:

100%



  loss 0.0490067131319
  err  0.0113345286885
  (0:00:07)
Epoch 379:

100%



  loss 0.0511855473195
  err  0.01171875
  (0:00:07)
Epoch 380:

100%



  loss 0.0486208262537
  err  0.0114839480874
  (0:00:07)
Epoch 381:

100%



  loss 0.0487564433197
  err  0.0104807035519
  (0:00:07)
Epoch 382:

100%



  loss 0.049960513631
  err  0.0117827868852
  (0:00:07)
Epoch 383:

100%



  loss 0.0504022942019
  err  0.0116547131148
  (0:00:07)
Epoch 384:

100%



  loss 0.0498614485968
  err  0.0118468237705
  (0:00:07)
Epoch 385:

100%



  loss 0.0500653735305
  err  0.0118681693989
  (0:00:07)
Epoch 386:

100%



  loss 0.049309084774
  err  0.011462602459
  (0:00:07)
Epoch 387:

100%



  loss 0.0526350030798
  err  0.012487192623
  (0:00:07)
Epoch 388:

100%



  loss 0.0501235535844
  err  0.0122737363388
  (0:00:07)
Epoch 389:

100%



  loss 0.0484625058429
  err  0.011206454918
  (0:00:07)
Epoch 390:

100%



  loss 0.0498601240494
  err  0.0112491461749
  (0:00:07)
Epoch 391:

100%



  loss 0.0507518351581
  err  0.0118895150273
  (0:00:07)
Epoch 392:

100%



  loss 0.0501922666848
  err  0.0117614412568
  (0:00:07)
Epoch 393:

100%



  loss 0.0508158842885
  err  0.0120389344262
  (0:00:07)
Epoch 394:

100%



  loss 0.0503427962877
  err  0.0111424180328
  (0:00:07)
Epoch 395:

100%



  loss 0.0512372959314
  err  0.0121243169399
  (0:00:07)
Epoch 396:

100%



  loss 0.0494296498669
  err  0.0115266393443
  (0:00:07)
Epoch 397:

100%



  loss 0.0504103448608
  err  0.0117614412568
  (0:00:07)
Epoch 398:

100%



  loss 0.0504657463145
  err  0.0115906762295
  (0:00:07)
Epoch 399:

100%



  loss 0.0506969279918
  err  0.0116333674863
  (0:00:07)
Epoch 400:

100%



  loss 0.0512759418588
  err  0.0121670081967
  (0:00:07)
Epoch 401:

100%



  loss 0.0517012247068
  err  0.0121883538251
  (0:00:07)
Epoch 402:

100%



  loss 0.0501304883332
  err  0.0116760587432
  (0:00:07)
Epoch 403:

100%



  loss 0.0487099933022
  err  0.011206454918
  (0:00:07)
Epoch 404:

100%



  loss 0.0482619690691
  err  0.0112278005464
  (0:00:07)
Epoch 405:

100%



  loss 0.0495880175283
  err  0.0113131830601
  (0:00:07)
Epoch 406:

100%



  loss 0.0497073170435
  err  0.0110356898907
  (0:00:07)
Epoch 407:

100%



  loss 0.0520832473374
  err  0.0117400956284
  (0:00:07)
Epoch 408:

100%



  loss 0.0484551596977
  err  0.0112918374317
  (0:00:07)
Epoch 409:

100%



  loss 0.04798519862
  err  0.0117827868852
  (0:00:07)
Epoch 410:

100%



  loss 0.0470664386594
  err  0.0108862704918
  (0:00:07)
Epoch 411:

100%



  loss 0.0497579451719
  err  0.0116333674863
  (0:00:07)
Epoch 412:

100%



  loss 0.0489679494621
  err  0.01171875
  (0:00:07)
Epoch 413:

100%



  loss 0.050957498675
  err  0.0117614412568
  (0:00:07)
Epoch 414:

100%



  loss 0.0501315662971
  err  0.0114199112022
  (0:00:07)
Epoch 415:

100%



  loss 0.047615717106
  err  0.0104166666667
  (0:00:07)
Epoch 416:

100%



  loss 0.0502720096052
  err  0.0113345286885
  (0:00:07)
Epoch 417:

100%



  loss 0.0530107693334
  err  0.0133196721311
  (0:00:07)
Epoch 418:

100%



  loss 0.0504115167337
  err  0.0119962431694
  (0:00:07)
Epoch 419:

100%



  loss 0.0504200194855
  err  0.0109076161202
  (0:00:07)
Epoch 420:

100%



  loss 0.050999305218
  err  0.0116974043716
  (0:00:07)
Epoch 421:

100%



  loss 0.0517370765539
  err  0.0125725751366
  (0:00:07)
Epoch 422:

100%



  loss 0.0500057769171
  err  0.0117827868852
  (0:00:07)
Epoch 423:

100%



  loss 0.052366873418
  err  0.0125512295082
  (0:00:07)
Epoch 424:

100%



  loss 0.0514989609714
  err  0.0121243169399
  (0:00:07)
Epoch 425:

100%



  loss 0.0495031032905
  err  0.0112918374317
  (0:00:07)
Epoch 426:

100%



  loss 0.0503172961667
  err  0.0119322062842
  (0:00:07)
Epoch 427:

100%



  loss 0.0492703765364
  err  0.0116760587432
  (0:00:07)
Epoch 428:

100%



  loss 0.0515410618092
  err  0.0118895150273
  (0:00:07)
Epoch 429:

100%



  loss 0.0502365323717
  err  0.0119322062842
  (0:00:07)
Epoch 430:

100%



  loss 0.0477823196412
  err  0.011099726776
  (0:00:07)
Epoch 431:

100%



  loss 0.04975405222
  err  0.0112918374317
  (0:00:07)
Epoch 432:

100%



  loss 0.0516214376962
  err  0.0116760587432
  (0:00:07)
Epoch 433:

100%



  loss 0.0500778241867
  err  0.0116547131148
  (0:00:07)
Epoch 434:

100%



  loss 0.0517748356396
  err  0.012487192623
  (0:00:07)
Epoch 435:

100%



  loss 0.0512037000853
  err  0.0118681693989
  (0:00:07)
Epoch 436:

100%



  loss 0.050197573581
  err  0.0116333674863
  (0:00:07)
Epoch 437:

100%



  loss 0.0516560903523
  err  0.0124445013661
  (0:00:07)
Epoch 438:

100%



  loss 0.0498092154008
  err  0.0120175887978
  (0:00:07)
Epoch 439:

100%



  loss 0.0493977603209
  err  0.0112491461749
  (0:00:07)
Epoch 440:

100%



  loss 0.050666761916
  err  0.0109716530055
  (0:00:07)
Epoch 441:

100%



  loss 0.0488958144446
  err  0.0112704918033
  (0:00:07)
Epoch 442:

100%



  loss 0.0479657469798
  err  0.0116120218579
  (0:00:07)
Epoch 443:

100%



  loss 0.0492923089642
  err  0.0112704918033
  (0:00:07)
Epoch 444:

100%



  loss 0.0496597786541
  err  0.0118254781421
  (0:00:07)
Epoch 445:

100%



  loss 0.0487362739479
  err  0.0116760587432
  (0:00:07)
Epoch 446:

100%



  loss 0.0495369232354
  err  0.0119535519126
  (0:00:07)
Epoch 447:

100%



  loss 0.050885064563
  err  0.011462602459
  (0:00:07)
Epoch 448:

100%



  loss 0.0512045377463
  err  0.0110570355191
  (0:00:07)
Epoch 449:

100%



  loss 0.050808559399
  err  0.0118681693989
  (0:00:07)
Epoch 450:

100%



  loss 0.0509543300713
  err  0.0113558743169
  (0:00:07)
Epoch 451:

100%



  loss 0.0524090463983
  err  0.012337773224
  (0:00:07)
Epoch 452:

100%



  loss 0.0488519621765
  err  0.010587431694
  (0:00:07)
Epoch 453:

100%



  loss 0.0490034788878
  err  0.011462602459
  (0:00:07)
Epoch 454:

100%



  loss 0.0506793434854
  err  0.0113345286885
  (0:00:07)
Epoch 455:

100%



  loss 0.0501720156594
  err  0.0124231557377
  (0:00:07)
Epoch 456:

100%



  loss 0.0495945964208
  err  0.0118895150273
  (0:00:07)
Epoch 457:

100%



  loss 0.0512701127935
  err  0.0120389344262
  (0:00:07)
Epoch 458:

100%



  loss 0.050681788744
  err  0.0118041325137
  (0:00:07)
Epoch 459:

100%



  loss 0.0496685736128
  err  0.011099726776
  (0:00:07)
Epoch 460:

100%



  loss 0.0509005833183
  err  0.0116760587432
  (0:00:07)
Epoch 461:

100%



  loss 0.0501581368621
  err  0.0118681693989
  (0:00:07)
Epoch 462:

100%



  loss 0.0490641129309
  err  0.0116120218579
  (0:00:07)
Epoch 463:

100%



  loss 0.0503530140249
  err  0.0122096994536
  (0:00:07)
Epoch 464:

100%



  loss 0.0498464214558
  err  0.0117400956284
  (0:00:07)
Epoch 465:

100%



  loss 0.0496122079298
  err  0.011462602459
  (0:00:07)
Epoch 466:

100%



  loss 0.0504250501924
  err  0.0116547131148
  (0:00:07)
Epoch 467:

100%



  loss 0.0509126304357
  err  0.0121883538251
  (0:00:07)
Epoch 468:

100%



  loss 0.049093916378
  err  0.0113345286885
  (0:00:07)
Epoch 469:

100%



  loss 0.0508008613512
  err  0.0118681693989
  (0:00:07)
Epoch 470:

100%



  loss 0.0515350075833
  err  0.0120816256831
  (0:00:07)
Epoch 471:

100%



  loss 0.0498636857189
  err  0.0115906762295
  (0:00:07)
Epoch 472:

100%



  loss 0.0504791550389
  err  0.0113345286885
  (0:00:07)
Epoch 473:

100%



  loss 0.0474447103101
  err  0.0114412568306
  (0:00:07)
Epoch 474:

100%



  loss 0.0500292390327
  err  0.0112918374317
  (0:00:07)
Epoch 475:

100%



  loss 0.0492814745038
  err  0.0113772199454
  (0:00:07)
Epoch 476:

100%



  loss 0.0500528491064
  err  0.0120175887978
  (0:00:07)
Epoch 477:

100%



  loss 0.0483490383269
  err  0.0111210724044
  (0:00:07)
Epoch 478:

100%



  loss 0.0494524344192
  err  0.0115906762295
  (0:00:07)
Epoch 479:

100%



  loss 0.0510079650434
  err  0.0118468237705
  (0:00:07)
Epoch 480:

100%



  loss 0.0506764951562
  err  0.0115693306011
  (0:00:07)
Epoch 481:

100%



  loss 0.0523215603685
  err  0.0122737363388
  (0:00:07)
Epoch 482:

100%



  loss 0.0496235628647
  err  0.0115266393443
  (0:00:07)
Epoch 483:

100%



  loss 0.0502154208921
  err  0.012337773224
  (0:00:07)
Epoch 484:

100%



  loss 0.0505185021632
  err  0.0117827868852
  (0:00:07)
Epoch 485:

100%



  loss 0.0493171139858
  err  0.0109716530055
  (0:00:07)
Epoch 486:

100%



  loss 0.0513237265112
  err  0.0119962431694
  (0:00:07)
Epoch 487:

100%



  loss 0.048204808975
  err  0.0112918374317
  (0:00:07)
Epoch 488:

100%



  loss 0.0495117197069
  err  0.0119108606557
  (0:00:07)
Epoch 489:

100%



  loss 0.0506332521982
  err  0.012487192623
  (0:00:07)
Epoch 490:

100%



  loss 0.0506829942451
  err  0.0126152663934
  (0:00:07)
Epoch 491:

100%



  loss 0.0483039389237
  err  0.0114199112022
  (0:00:07)
Epoch 492:

100%



  loss 0.0495354605882
  err  0.0113131830601
  (0:00:07)
Epoch 493:

100%



  loss 0.0500667936791
  err  0.0121243169399
  (0:00:07)
Epoch 494:

100%



  loss 0.0497409211194
  err  0.0113131830601
  (0:00:07)
Epoch 495:

100%



  loss 0.0505902965126
  err  0.0112704918033
  (0:00:07)
Epoch 496:

100%



  loss 0.0504925284443
  err  0.0117400956284
  (0:00:07)
Epoch 497:

100%



  loss 0.0499645343421
  err  0.011974897541
  (0:00:07)
Epoch 498:

100%



  loss 0.048930592688
  err  0.011206454918
  (0:00:07)
Epoch 499:

100%



  loss 0.0496774319619
  err  0.010843579235
  (0:00:07)
Epoch 500:

100%



  loss 0.0485952898359
  err  0.010843579235
  (0:00:07)
Epoch 501:

100%



  loss 0.0480258661727
  err  0.0114199112022
  (0:00:07)
Epoch 502:

100%



  loss 0.0488127943292
  err  0.0113131830601
  (0:00:07)
Epoch 503:

100%



  loss 0.0486774558769
  err  0.0114839480874
  (0:00:07)
Epoch 504:

100%



  loss 0.0491050281409
  err  0.0114412568306
  (0:00:07)
Epoch 505:

100%



  loss 0.05055996344
  err  0.012231045082
  (0:00:07)
Epoch 506:

100%



  loss 0.0530303752278
  err  0.0125512295082
  (0:00:07)
Epoch 507:

100%



  loss 0.052891744101
  err  0.0121243169399
  (0:00:07)
Epoch 508:

100%



  loss 0.0505175959615
  err  0.011462602459
  (0:00:07)
Epoch 509:

100%



  loss 0.0494136192653
  err  0.0118895150273
  (0:00:07)
Epoch 510:

100%



  loss 0.0522509685436
  err  0.0117614412568
  (0:00:07)
Epoch 511:

100%



  loss 0.0509286801664
  err  0.0124658469945
  (0:00:07)
Epoch 512:

100%



  loss 0.0506521595656
  err  0.0120175887978
  (0:00:07)
Epoch 513:

100%



  loss 0.0498278064996
  err  0.0116974043716
  (0:00:07)
Epoch 514:

100%



  loss 0.0486504111561
  err  0.0115479849727
  (0:00:07)
Epoch 515:

100%



  loss 0.0511386389755
  err  0.0121029713115
  (0:00:07)
Epoch 516:

100%



  loss 0.0511584906251
  err  0.0115479849727
  (0:00:07)
Epoch 517:

100%



  loss 0.0495171647136
  err  0.01171875
  (0:00:07)
Epoch 518:

100%



  loss 0.0514544843972
  err  0.0122096994536
  (0:00:07)
Epoch 519:

100%



  loss 0.0497379028467
  err  0.0112704918033
  (0:00:07)
Epoch 520:

100%



  loss 0.0526834636039
  err  0.012487192623
  (0:00:07)
Epoch 521:

100%



  loss 0.051112405784
  err  0.0124018101093
  (0:00:07)
Epoch 522:

100%



  loss 0.0496806453868
  err  0.0109289617486
  (0:00:07)
Epoch 523:

100%



  loss 0.0485191454817
  err  0.0111637636612
  (0:00:07)
Epoch 524:

100%



  loss 0.0489324341691
  err  0.0110143442623
  (0:00:07)
Epoch 525:

100%



  loss 0.0521278063477
  err  0.0128927595628
  (0:00:07)
Epoch 526:

100%



  loss 0.0491814250553
  err  0.0113985655738
  (0:00:07)
Epoch 527:

100%



  loss 0.0497417541022
  err  0.0122523907104
  (0:00:07)
Epoch 528:

100%



  loss 0.0508460697904
  err  0.0120389344262
  (0:00:07)
Epoch 529:

100%



  loss 0.0512529746349
  err  0.0119322062842
  (0:00:07)
Epoch 530:

100%



  loss 0.0486017193676
  err  0.0115266393443
  (0:00:07)
Epoch 531:

100%



  loss 0.0495011471963
  err  0.0117614412568
  (0:00:07)
Epoch 532:

100%



  loss 0.0490276832833
  err  0.0118681693989
  (0:00:07)
Epoch 533:

100%



  loss 0.0491637419591
  err  0.011206454918
  (0:00:07)
Epoch 534:

100%



  loss 0.0507331994977
  err  0.0116974043716
  (0:00:07)
Epoch 535:

100%



  loss 0.048799170921
  err  0.0111637636612
  (0:00:07)
Epoch 536:

100%



  loss 0.0509425331202
  err  0.0122950819672
  (0:00:07)
Epoch 537:

100%



  loss 0.0526768281875
  err  0.0120816256831
  (0:00:07)
Epoch 538:

100%



  loss 0.0497858111729
  err  0.0118468237705
  (0:00:07)
Epoch 539:

100%



  loss 0.0494644982152
  err  0.0120602800546
  (0:00:07)
Epoch 540:

100%



  loss 0.0515113341249
  err  0.011974897541
  (0:00:07)
Epoch 541:

100%



  loss 0.0493378850989
  err  0.0117827868852
  (0:00:07)
Epoch 542:

100%



  loss 0.0492579228247
  err  0.0118041325137
  (0:00:07)
Epoch 543:

100%



  loss 0.0500463201248
  err  0.0112278005464
  (0:00:07)
Epoch 544:

100%



  loss 0.0504576274266
  err  0.0116333674863
  (0:00:07)
Epoch 545:

100%



  loss 0.0498696480957
  err  0.0113131830601
  (0:00:07)
Epoch 546:

100%



  loss 0.0484162926307
  err  0.0116333674863
  (0:00:07)
Epoch 547:

100%



  loss 0.0488542137003
  err  0.0115906762295
  (0:00:07)
Epoch 548:

100%



  loss 0.0500817800582
  err  0.0113345286885
  (0:00:07)
Epoch 549:

100%



  loss 0.0498693933618
  err  0.0115906762295
  (0:00:07)
Epoch 550:

100%



  loss 0.0490234316711
  err  0.0115479849727
  (0:00:07)
Epoch 551:

100%



  loss 0.0490403627898
  err  0.0115479849727
  (0:00:07)
Epoch 552:

100%



  loss 0.0489942923416
  err  0.0113772199454
  (0:00:07)
Epoch 553:

100%



  loss 0.0508350694757
  err  0.0115906762295
  (0:00:07)
Epoch 554:

100%



  loss 0.0481245872613
  err  0.0114412568306
  (0:00:07)
Epoch 555:

100%



  loss 0.051482576443
  err  0.0120816256831
  (0:00:07)
Epoch 556:

100%



  loss 0.0502144057931
  err  0.0118254781421
  (0:00:07)
Epoch 557:

100%



  loss 0.0505547662173
  err  0.0115906762295
  (0:00:07)
Epoch 558:

100%



  loss 0.0512593156427
  err  0.011974897541
  (0:00:07)
Epoch 559:

100%



  loss 0.0493645557244
  err  0.0116120218579
  (0:00:07)
Epoch 560:

100%



  loss 0.0488224675881
  err  0.0113558743169
  (0:00:07)
Epoch 561:

100%



  loss 0.0489594621827
  err  0.0106301229508
  (0:00:07)
Epoch 562:

100%



  loss 0.0493275828264
  err  0.0112704918033
  (0:00:07)
Epoch 563:

100%



  loss 0.0524742716104
  err  0.0117400956284
  (0:00:07)
Epoch 564:

100%



  loss 0.0497446582421
  err  0.0108862704918
  (0:00:07)
Epoch 565:

100%



  loss 0.0504000849152
  err  0.0122950819672
  (0:00:07)
Epoch 566:

100%



  loss 0.0489448801203
  err  0.0113345286885
  (0:00:07)
Epoch 567:

100%



  loss 0.0499624332414
  err  0.01171875
  (0:00:07)
Epoch 568:

100%



  loss 0.0514960068525
  err  0.0121456625683
  (0:00:07)
Epoch 569:

100%



  loss 0.0514319095871
  err  0.012231045082
  (0:00:07)
Epoch 570:

100%



  loss 0.0520980874219
  err  0.0125725751366
  (0:00:07)
Epoch 571:

100%



  loss 0.051654903825
  err  0.0124658469945
  (0:00:07)
Epoch 572:

100%



  loss 0.0501975743991
  err  0.0116760587432
  (0:00:07)
Epoch 573:

100%



  loss 0.0508982122079
  err  0.0118468237705
  (0:00:07)
Epoch 574:

100%



  loss 0.0490594221752
  err  0.0114412568306
  (0:00:07)
Epoch 575:

100%



  loss 0.0517845237895
  err  0.0126579576503
  (0:00:07)
Epoch 576:

100%



  loss 0.0487323995194
  err  0.0111424180328
  (0:00:07)
Epoch 577:

100%



  loss 0.0480324140193
  err  0.0103526297814
  (0:00:07)
Epoch 578:

100%



  loss 0.0506722678461
  err  0.011974897541
  (0:00:07)
Epoch 579:

100%



  loss 0.0497474704762
  err  0.0118468237705
  (0:00:07)
Epoch 580:

100%



  loss 0.0488731874117
  err  0.0114199112022
  (0:00:07)
Epoch 581:

100%



  loss 0.0522129736838
  err  0.0123164275956
  (0:00:07)
Epoch 582:

100%



  loss 0.0506881782753
  err  0.0115906762295
  (0:00:07)
Epoch 583:

100%



  loss 0.0489221462876
  err  0.0112278005464
  (0:00:07)
Epoch 584:

100%



  loss 0.04908525041
  err  0.0110783811475
  (0:00:07)
Epoch 585:

100%



  loss 0.0516544734815
  err  0.012487192623
  (0:00:07)
Epoch 586:

100%



  loss 0.049545381222
  err  0.0116547131148
  (0:00:07)
Epoch 587:

100%



  loss 0.050272196635
  err  0.0116974043716
  (0:00:07)
Epoch 588:

100%



  loss 0.0494660405319
  err  0.0113345286885
  (0:00:07)
Epoch 589:

100%



  loss 0.0484222261969
  err  0.0115052937158
  (0:00:07)
Epoch 590:

100%



  loss 0.0495990016504
  err  0.0116333674863
  (0:00:07)
Epoch 591:

100%



  loss 0.050832080679
  err  0.0121029713115
  (0:00:07)
Epoch 592:

100%



  loss 0.0509058064291
  err  0.0119108606557
  (0:00:07)
Epoch 593:

100%



  loss 0.0494550355987
  err  0.0116333674863
  (0:00:07)
Epoch 594:

100%



  loss 0.0482766247645
  err  0.0111210724044
  (0:00:07)
Epoch 595:

100%



  loss 0.0525472967541
  err  0.0128073770492
  (0:00:07)
Epoch 596:

100%



  loss 0.0485393270205
  err  0.0113772199454
  (0:00:07)
Epoch 597:

100%



  loss 0.0505575821121
  err  0.011974897541
  (0:00:07)
Epoch 598:

100%



  loss 0.0519198403984
  err  0.0121243169399
  (0:00:07)
Epoch 599:

100%



  loss 0.0497809115268
  err  0.0115479849727
  (0:00:07)
Epoch 600:

100%



  loss 0.0534107082592
  err  0.0125725751366
  (0:00:07)
Epoch 601:

100%



  loss 0.0509371384889
  err  0.0119322062842
  (0:00:07)
Epoch 602:

100%



  loss 0.0524013240674
  err  0.0125512295082
  (0:00:07)
Epoch 603:

100%



  loss 0.0511411581802
  err  0.0121670081967
  (0:00:07)
Epoch 604:

100%



  loss 0.0496994043176
  err  0.0113345286885
  (0:00:07)
Epoch 605:

100%



  loss 0.0510246200892
  err  0.0119535519126
  (0:00:07)
Epoch 606:

100%



  loss 0.0505830776735
  err  0.0113558743169
  (0:00:07)
Epoch 607:

100%



  loss 0.0513388772077
  err  0.01171875
  (0:00:07)
Epoch 608:

100%



  loss 0.0505572161179
  err  0.0118468237705
  (0:00:07)
Epoch 609:

100%



  loss 0.0500303413363
  err  0.0117827868852
  (0:00:07)
Epoch 610:

100%



  loss 0.0519695952713
  err  0.0127006489071
  (0:00:07)
Epoch 611:

100%



  loss 0.0509097522272
  err  0.0118254781421
  (0:00:07)
Epoch 612:

100%



  loss 0.0476686004613
  err  0.0109716530055
  (0:00:07)
Epoch 613:

100%



  loss 0.0493728377623
  err  0.0116760587432
  (0:00:07)
Epoch 614:

100%



  loss 0.048385312497
  err  0.0115052937158
  (0:00:07)
Epoch 615:

100%



  loss 0.051595519384
  err  0.0122096994536
  (0:00:07)
Epoch 616:

100%



  loss 0.0500790580784
  err  0.012487192623
  (0:00:07)
Epoch 617:

100%



  loss 0.0501288911277
  err  0.0121456625683
  (0:00:07)
Epoch 618:

100%



  loss 0.0485964923273
  err  0.0111637636612
  (0:00:07)
Epoch 619:

100%



  loss 0.0492568608791
  err  0.0113131830601
  (0:00:07)
Epoch 620:

100%



  loss 0.0498558964944
  err  0.0112278005464
  (0:00:07)
Epoch 621:

100%



  loss 0.0490498096552
  err  0.0110570355191
  (0:00:07)
Epoch 622:

100%



  loss 0.0508856944711
  err  0.0120602800546
  (0:00:07)
Epoch 623:

100%



  loss 0.0522979684543
  err  0.0116120218579
  (0:00:07)
Epoch 624:

100%



  loss 0.0532817443657
  err  0.0130421789617
  (0:00:07)
Epoch 625:

100%



  loss 0.0522336068587
  err  0.0119535519126
  (0:00:07)
Epoch 626:

100%



  loss 0.0515883239527
  err  0.0120175887978
  (0:00:07)
Epoch 627:

100%



  loss 0.0494901981177
  err  0.0121029713115
  (0:00:07)
Epoch 628:

100%



  loss 0.0521485246985
  err  0.0125512295082
  (0:00:07)
Epoch 629:

100%



  loss 0.0512299289357
  err  0.0116333674863
  (0:00:07)
Epoch 630:

100%



  loss 0.0506708166967
  err  0.0120175887978
  (0:00:07)
Epoch 631:

100%



  loss 0.0513034970919
  err  0.0115906762295
  (0:00:07)
Epoch 632:

100%



  loss 0.0504935263743
  err  0.0119322062842
  (0:00:07)
Epoch 633:

100%



  loss 0.049465872542
  err  0.0111637636612
  (0:00:07)
Epoch 634:

100%



  loss 0.0496725490383
  err  0.0118895150273
  (0:00:07)
Epoch 635:

100%



  loss 0.0500824092436
  err  0.0117400956284
  (0:00:07)
Epoch 636:

100%



  loss 0.0496909950867
  err  0.011462602459
  (0:00:07)
Epoch 637:

100%



  loss 0.0486297786943
  err  0.0115052937158
  (0:00:07)
Epoch 638:

100%



  loss 0.0498161165736
  err  0.0113558743169
  (0:00:07)
Epoch 639:

100%



  loss 0.0521146288145
  err  0.0124658469945
  (0:00:07)
Epoch 640:

100%



  loss 0.049090654497
  err  0.0108008879781
  (0:00:07)
Epoch 641:

100%



  loss 0.0496594112486
  err  0.0118895150273
  (0:00:07)
Epoch 642:

100%



  loss 0.0496459415646
  err  0.0120816256831
  (0:00:07)
Epoch 643:

100%



  loss 0.0492744025891
  err  0.0113345286885
  (0:00:07)
Epoch 644:

100%



  loss 0.0503596462197
  err  0.0119322062842
  (0:00:07)
Epoch 645:

100%



  loss 0.0515378265033
  err  0.013106215847
  (0:00:07)
Epoch 646:

100%



  loss 0.0516632638485
  err  0.0120389344262
  (0:00:07)
Epoch 647:

100%



  loss 0.0507366384775
  err  0.0114839480874
  (0:00:07)
Epoch 648:

100%



  loss 0.0509215150973
  err  0.0122096994536
  (0:00:07)
Epoch 649:

100%



  loss 0.0507965800212
  err  0.0119962431694
  (0:00:08)
Epoch 650:

100%



  loss 0.0489765348879
  err  0.0113558743169
  (0:00:08)
Epoch 651:

100%



  loss 0.0517149995376
  err  0.0126366120219
  (0:00:08)
Epoch 652:

100%



  loss 0.0476750508867
  err  0.0106941598361
  (0:00:08)
Epoch 653:

100%



  loss 0.0505961095558
  err  0.0120389344262
  (0:00:08)
Epoch 654:

100%



  loss 0.0502864234035
  err  0.0116974043716
  (0:00:08)
Epoch 655:

100%



  loss 0.0499947249006
  err  0.0113131830601
  (0:00:08)
Epoch 656:

100%



  loss 0.0490841477953
  err  0.0112918374317
  (0:00:08)
Epoch 657:

100%



  loss 0.049938226106
  err  0.0115052937158
  (0:00:08)
Epoch 658:

100%



  loss 0.0465028554779
  err  0.0107581967213
  (0:00:08)
Epoch 659:

100%



  loss 0.0502019398948
  err  0.0112278005464
  (0:00:08)
Epoch 660:

100%



  loss 0.0507503025274
  err  0.0115906762295
  (0:00:08)
Epoch 661:

100%



  loss 0.0513190257724
  err  0.0123591188525
  (0:00:08)
Epoch 662:

100%



  loss 0.0487885927664
  err  0.0111637636612
  (0:00:08)
Epoch 663:

100%



  loss 0.0513905346727
  err  0.0121670081967
  (0:00:09)
Epoch 664:

100%



  loss 0.0486509857707
  err  0.0113772199454
  (0:00:09)
Epoch 665:

100%



  loss 0.0500016853327
  err  0.0117614412568
  (0:00:09)
Epoch 666:

100%



  loss 0.0492992741705
  err  0.0117827868852
  (0:00:09)
Epoch 667:

100%



  loss 0.0525121252425
  err  0.0123804644809
  (0:00:09)
Epoch 668:

100%



  loss 0.0493247968309
  err  0.0116120218579
  (0:00:09)
Epoch 669:

100%



  loss 0.0510224788889
  err  0.0120602800546
  (0:00:09)
Epoch 670:

100%



  loss 0.0528118516671
  err  0.0120175887978
  (0:00:09)
Epoch 671:

100%



  loss 0.0504957784764
  err  0.0118895150273
  (0:00:09)
Epoch 672:

100%



  loss 0.0493391565737
  err  0.0115266393443
  (0:00:09)
Epoch 673:

100%



  loss 0.0512829998666
  err  0.0118468237705
  (0:00:09)
Epoch 674:

100%



  loss 0.0501135381771
  err  0.0117614412568
  (0:00:10)
Epoch 675:

100%



  loss 0.0492136629344
  err  0.0115693306011
  (0:00:10)
Epoch 676:

100%



  loss 0.0485449126156
  err  0.0109076161202
  (0:00:10)
Epoch 677:

100%



  loss 0.0502702731778
  err  0.0116333674863
  (0:00:10)
Epoch 678:

100%



  loss 0.0502685698581
  err  0.0115906762295
  (0:00:10)
Epoch 679:

100%



  loss 0.0495147602423
  err  0.0121243169399
  (0:00:10)
Epoch 680:

100%



  loss 0.0505789234837
  err  0.0120816256831
  (0:00:10)
Epoch 681:

100%



  loss 0.0503835738611
  err  0.0118681693989
  (0:00:10)
Epoch 682:

100%



  loss 0.0505135277212
  err  0.0116333674863
  (0:00:10)
Epoch 683:

100%



  loss 0.0494064111848
  err  0.0110143442623
  (0:00:11)
Epoch 684:

100%



  loss 0.0523768982696
  err  0.0121029713115
  (0:00:11)
Epoch 685:

100%



  loss 0.0481479461903
  err  0.0112491461749
  (0:00:11)
Epoch 686:

100%



  loss 0.0496807244875
  err  0.011462602459
  (0:00:11)
Epoch 687:

100%



  loss 0.0488677610258
  err  0.0108649248634
  (0:00:11)
Epoch 688:

100%



  loss 0.0503486328188
  err  0.0115693306011
  (0:00:11)
Epoch 689:

100%



  loss 0.050189324292
  err  0.0113772199454
  (0:00:11)
Epoch 690:

100%



  loss 0.0499997654277
  err  0.0116760587432
  (0:00:11)
Epoch 691:

100%



  loss 0.0495655755601
  err  0.0117827868852
  (0:00:11)
Epoch 692:

100%



  loss 0.0482603266366
  err  0.0111637636612
  (0:00:11)
Epoch 693:

100%



  loss 0.0514813014204
  err  0.0118254781421
  (0:00:11)
Epoch 694:

100%



  loss 0.0500286214163
  err  0.0114199112022
  (0:00:11)
Epoch 695:

100%



  loss 0.0509122768996
  err  0.0121883538251
  (0:00:11)
Epoch 696:

100%



  loss 0.0505350526165
  err  0.011974897541
  (0:00:11)
Epoch 697:

100%



  loss 0.0494034524436
  err  0.0115052937158
  (0:00:11)
Epoch 698:

100%



  loss 0.0495845049154
  err  0.0118254781421
  (0:00:11)
Epoch 699:

100%



  loss 0.0515384797994
  err  0.0123591188525
  (0:00:11)
Epoch 700:

100%



  loss 0.0509073899759
  err  0.0122950819672
  (0:00:11)
Epoch 701:

100%



  loss 0.0500637556081
  err  0.0113772199454
  (0:00:11)
Epoch 702:

100%



  loss 0.0519052749332
  err  0.0126152663934
  (0:00:11)
Epoch 703:

100%



  loss 0.0502056034773
  err  0.0116120218579
  (0:00:11)
Epoch 704:

100%



  loss 0.05061755868
  err  0.0114839480874
  (0:00:11)
Epoch 705:

100%



  loss 0.0513489583165
  err  0.0123591188525
  (0:00:11)
Epoch 706:

100%



  loss 0.0478177824436
  err  0.0111637636612
  (0:00:12)
Epoch 707:

100%



  loss 0.0507488015885
  err  0.0119535519126
  (0:00:12)
Epoch 708:

100%



  loss 0.0500263936221
  err  0.0119962431694
  (0:00:12)
Epoch 709:

100%



  loss 0.0492875696391
  err  0.0119108606557
  (0:00:12)
Epoch 710:

100%



  loss 0.0496841880106
  err  0.0113772199454
  (0:00:12)
Epoch 711:

100%



  loss 0.0496682745705
  err  0.0111637636612
  (0:00:12)
Epoch 712:

100%



  loss 0.0503036967896
  err  0.011462602459
  (0:00:12)
Epoch 713:

100%



  loss 0.0501081825315
  err  0.0119962431694
  (0:00:12)
Epoch 714:

100%



  loss 0.0483309057658
  err  0.0111851092896
  (0:00:12)
Epoch 715:

100%



  loss 0.0497137271962
  err  0.0113345286885
  (0:00:12)
Epoch 716:

100%



  loss 0.0483101033046
  err  0.0112491461749
  (0:00:12)
Epoch 717:

100%



  loss 0.0492212572033
  err  0.0109716530055
  (0:00:12)
Epoch 718:

100%



  loss 0.0510782047361
  err  0.0119535519126
  (0:00:12)
Epoch 719:

100%



  loss 0.0507845541989
  err  0.0115693306011
  (0:00:12)
Epoch 720:

100%



  loss 0.0519552215606
  err  0.0123164275956
  (0:00:12)
Epoch 721:

100%



  loss 0.0498332024051
  err  0.0115906762295
  (0:00:12)
Epoch 722:

100%



  loss 0.0514724525252
  err  0.0119535519126
  (0:00:12)
Epoch 723:

100%



  loss 0.0506799651693
  err  0.0111637636612
  (0:00:12)
Epoch 724:

100%



  loss 0.0516177981792
  err  0.012593920765
  (0:00:12)
Epoch 725:

100%



  loss 0.0511634300152
  err  0.0121243169399
  (0:00:12)
Epoch 726:

100%



  loss 0.0515765273788
  err  0.0123164275956
  (0:00:12)
Epoch 727:

100%



  loss 0.049406080695
  err  0.0112278005464
  (0:00:12)
Epoch 728:

100%



  loss 0.0518597290429
  err  0.0125725751366
  (0:00:12)
Epoch 729:

100%



  loss 0.0486239953834
  err  0.0110143442623
  (0:00:12)
Epoch 730:

100%



  loss 0.0509860216159
  err  0.0123164275956
  (0:00:12)
Epoch 731:

100%



  loss 0.0508695249013
  err  0.0123591188525
  (0:00:12)
Epoch 732:

100%



  loss 0.0494847734404
  err  0.0110783811475
  (0:00:12)
Epoch 733:

100%



  loss 0.0502947142947
  err  0.0115479849727
  (0:00:12)
Epoch 734:

100%



  loss 0.0515002316421
  err  0.0125298838798
  (0:00:12)
Epoch 735:

100%



  loss 0.0496758904818
  err  0.0118895150273
  (0:00:12)
Epoch 736:

100%



  loss 0.0526358996183
  err  0.0122950819672
  (0:00:12)
Epoch 737:

100%



  loss 0.049629877698
  err  0.0116760587432
  (0:00:12)
Epoch 738:

100%



  loss 0.0507310102178
  err  0.0115479849727
  (0:00:12)
Epoch 739:

100%



  loss 0.0511513752278
  err  0.0125725751366
  (0:00:12)
Epoch 740:

100%



  loss 0.0504448505372
  err  0.0121670081967
  (0:00:12)
Epoch 741:

100%



  loss 0.0485758004631
  err  0.0111851092896
  (0:00:12)
Epoch 742:

100%



  loss 0.0503104487701
  err  0.012487192623
  (0:00:12)
Epoch 743:

100%



  loss 0.049191763505
  err  0.011206454918
  (0:00:12)
Epoch 744:

100%



  loss 0.0507904036222
  err  0.0118468237705
  (0:00:12)
Epoch 745:

100%



  loss 0.0492462052078
  err  0.0112704918033
  (0:00:12)
Epoch 746:

100%



  loss 0.0497594571779
  err  0.0112278005464
  (0:00:12)
Epoch 747:

100%



  loss 0.0486606443587
  err  0.0116760587432
  (0:00:12)
Epoch 748:

100%



  loss 0.0517921192785
  err  0.0129567964481
  (0:00:12)
Epoch 749:

100%



  loss 0.0497257524944
  err  0.0119322062842
  (0:00:12)
Epoch 750:

100%



  loss 0.0491255763759
  err  0.0110143442623
  (0:00:12)
Epoch 751:

100%



  loss 0.0520221476972
  err  0.0118895150273
  (0:00:12)
Epoch 752:

100%



  loss 0.05055292417
  err  0.0117614412568
  (0:00:12)
Epoch 753:

100%



  loss 0.0486802094491
  err  0.0110783811475
  (0:00:12)
Epoch 754:

100%



  loss 0.0485195475045
  err  0.0109289617486
  (0:00:12)
Epoch 755:

100%



  loss 0.0492029296084
  err  0.0113345286885
  (0:00:12)
Epoch 756:

100%



  loss 0.0496815428452
  err  0.0116333674863
  (0:00:12)
Epoch 757:

100%



  loss 0.0493317350581
  err  0.0115693306011
  (0:00:12)
Epoch 758:

100%



  loss 0.0522228305869
  err  0.0118681693989
  (0:00:12)
Epoch 759:

100%



  loss 0.049665660703
  err  0.0114412568306
  (0:00:12)
Epoch 760:

100%



  loss 0.0504269773367
  err  0.0118041325137
  (0:00:12)
Epoch 761:

100%



  loss 0.0499248313495
  err  0.0119322062842
  (0:00:12)
Epoch 762:

100%



  loss 0.0491016525457
  err  0.0112704918033
  (0:00:12)
Epoch 763:

100%



  loss 0.048877800192
  err  0.0113345286885
  (0:00:12)
Epoch 764:

100%



  loss 0.0487663296854
  err  0.0117827868852
  (0:00:12)
Epoch 765:

100%



  loss 0.0487446565171
  err  0.0112491461749
  (0:00:12)
Epoch 766:

100%



  loss 0.0502527322292
  err  0.0118468237705
  (0:00:12)
Epoch 767:

100%



  loss 0.0509214801332
  err  0.0118468237705
  (0:00:12)
Epoch 768:

100%



  loss 0.0505621534419
  err  0.0115479849727
  (0:00:12)
Epoch 769:

100%



  loss 0.0496587947773
  err  0.0120602800546
  (0:00:12)
Epoch 770:

100%



  loss 0.0514970282462
  err  0.0122950819672
  (0:00:12)
Epoch 771:

100%



  loss 0.0487626440259
  err  0.0108008879781
  (0:00:12)
Epoch 772:

100%



  loss 0.0517847655892
  err  0.0118254781421
  (0:00:12)
Epoch 773:

100%



  loss 0.0492733478313
  err  0.0111851092896
  (0:00:12)
Epoch 774:

100%



  loss 0.0505736138719
  err  0.0120389344262
  (0:00:12)
Epoch 775:

100%



  loss 0.0502263119691
  err  0.0121883538251
  (0:00:12)
Epoch 776:

100%



  loss 0.0503167000146
  err  0.0115693306011
  (0:00:12)
Epoch 777:

100%



  loss 0.0504568956813
  err  0.0121670081967
  (0:00:12)
Epoch 778:

100%



  loss 0.0513021366828
  err  0.0120602800546
  (0:00:12)
Epoch 779:

100%



  loss 0.0471308205443
  err  0.0106728142077
  (0:00:12)
Epoch 780:

100%



  loss 0.0532776464268
  err  0.0127860314208
  (0:00:12)
Epoch 781:

100%



  loss 0.0489919872603
  err  0.0115479849727
  (0:00:12)
Epoch 782:

100%



  loss 0.0495732420498
  err  0.0108222336066
  (0:00:12)
Epoch 783:

100%



  loss 0.0475728117328
  err  0.0108222336066
  (0:00:12)
Epoch 784:

100%



  loss 0.0485415138416
  err  0.0116760587432
  (0:00:12)
Epoch 785:

100%



  loss 0.0504405921887
  err  0.0116974043716
  (0:00:12)
Epoch 786:

100%



  loss 0.0493150216634
  err  0.0113772199454
  (0:00:12)
Epoch 787:

100%



  loss 0.0521425744213
  err  0.0129141051913
  (0:00:12)
Epoch 788:

100%



  loss 0.0499034862777
  err  0.011974897541
  (0:00:12)
Epoch 789:

100%



  loss 0.0506525020805
  err  0.0123591188525
  (0:00:12)
Epoch 790:

100%



  loss 0.0503721266458
  err  0.0120389344262
  (0:00:12)
Epoch 791:

100%



  loss 0.0495341655778
  err  0.0115052937158
  (0:00:12)
Epoch 792:

100%



  loss 0.0504048328524
  err  0.0118468237705
  (0:00:12)
Epoch 793:

100%



  loss 0.0496526895348
  err  0.010950307377
  (0:00:12)
Epoch 794:

100%



  loss 0.0511541717808
  err  0.0122523907104
  (0:00:12)
Epoch 795:

100%



  loss 0.0508440480506
  err  0.0120389344262
  (0:00:12)
Epoch 796:

100%



  loss 0.0495411016981
  err  0.0113558743169
  (0:00:12)
Epoch 797:

100%



  loss 0.0489243634345
  err  0.0112704918033
  (0:00:12)
Epoch 798:

100%



  loss 0.0508952564763
  err  0.0115906762295
  (0:00:12)
Epoch 799:

100%



  loss 0.0501921483216
  err  0.0116760587432
  (0:00:12)
Epoch 800:

100%



  loss 0.0520041190123
  err  0.0127006489071
  (0:00:12)
Epoch 801:

100%



  loss 0.0496922919534
  err  0.011974897541
  (0:00:12)
Epoch 802:

100%



  loss 0.0512084310857
  err  0.0121456625683
  (0:00:12)
Epoch 803:

100%



  loss 0.0505210807333
  err  0.0120389344262
  (0:00:12)
Epoch 804:

100%



  loss 0.0505267252294
  err  0.0119322062842
  (0:00:12)
Epoch 805:

100%



  loss 0.050404978007
  err  0.0115693306011
  (0:00:12)
Epoch 806:

100%



  loss 0.0510058671343
  err  0.0119322062842
  (0:00:12)
Epoch 807:

100%



  loss 0.052394630736
  err  0.0117400956284
  (0:00:12)
Epoch 808:

100%



  loss 0.0514188094963
  err  0.01171875
  (0:00:12)
Epoch 809:

100%



  loss 0.0507988231403
  err  0.011974897541
  (0:00:12)
Epoch 810:

100%



  loss 0.0486905402523
  err  0.0115052937158
  (0:00:12)
Epoch 811:

100%



  loss 0.0514803659075
  err  0.0123804644809
  (0:00:12)
Epoch 812:

100%



  loss 0.0493847168403
  err  0.0118895150273
  (0:00:12)
Epoch 813:

100%



  loss 0.0485055574402
  err  0.0112918374317
  (0:00:12)
Epoch 814:

100%



  loss 0.0506335625481
  err  0.0119108606557
  (0:00:12)
Epoch 815:

100%



  loss 0.050800062118
  err  0.0119322062842
  (0:00:12)
Epoch 816:

100%



  loss 0.0488600122152
  err  0.0112918374317
  (0:00:12)
Epoch 817:

100%



  loss 0.0506314623731
  err  0.0119108606557
  (0:00:12)
Epoch 818:

100%



  loss 0.0516778285516
  err  0.0115479849727
  (0:00:12)
Epoch 819:

100%



  loss 0.0487951206761
  err  0.0111637636612
  (0:00:12)
Epoch 820:

100%



  loss 0.0494213312969
  err  0.0113772199454
  (0:00:12)
Epoch 821:

100%



  loss 0.0515666218586
  err  0.012487192623
  (0:00:12)
Epoch 822:

100%



  loss 0.0482091772576
  err  0.0113131830601
  (0:00:12)
Epoch 823:

100%



  loss 0.0497920159562
  err  0.0117400956284
  (0:00:12)
Epoch 824:

100%



  loss 0.0482130055456
  err  0.0108862704918
  (0:00:12)
Epoch 825:

100%



  loss 0.0509786604168
  err  0.0119108606557
  (0:00:12)
Epoch 826:

100%



  loss 0.0505859719442
  err  0.0115906762295
  (0:00:12)
Epoch 827:

100%



  loss 0.0507456185932
  err  0.0116760587432
  (0:00:12)
Epoch 828:

100%



  loss 0.0495457933323
  err  0.0116547131148
  (0:00:12)
Epoch 829:

100%



  loss 0.0491290222077
  err  0.0111851092896
  (0:00:12)
Epoch 830:

100%



  loss 0.0498125351065
  err  0.0116333674863
  (0:00:12)
Epoch 831:

100%



  loss 0.0519524026049
  err  0.0117400956284
  (0:00:13)
Epoch 832:

100%



  loss 0.0510293716607
  err  0.0124445013661
  (0:00:12)
Epoch 833:

100%



  loss 0.0499267532437
  err  0.0117400956284
  (0:00:12)
Epoch 834:

100%



  loss 0.0505288892751
  err  0.0121670081967
  (0:00:13)
Epoch 835:

100%



  loss 0.0482803951603
  err  0.0112491461749
  (0:00:12)
Epoch 836:

100%



  loss 0.0505963224851
  err  0.0115266393443
  (0:00:13)
Epoch 837:

100%



  loss 0.0502980490319
  err  0.0116760587432
  (0:00:12)
Epoch 838:

100%



  loss 0.0492441997879
  err  0.0112491461749
  (0:00:12)
Epoch 839:

100%



  loss 0.0509210809037
  err  0.0123164275956
  (0:00:12)
Epoch 840:

100%



  loss 0.0515115752847
  err  0.0120389344262
  (0:00:12)
Epoch 841:

100%



  loss 0.0503400050694
  err  0.0116120218579
  (0:00:12)
Epoch 842:

100%



  loss 0.0506105905817
  err  0.0121456625683
  (0:00:13)
Epoch 843:

100%



  loss 0.0485212640872
  err  0.0114839480874
  (0:00:12)
Epoch 844:

100%



  loss 0.0494375340157
  err  0.0115052937158
  (0:00:13)
Epoch 845:

100%



  loss 0.0488792511958
  err  0.0113345286885
  (0:00:15)
Epoch 846:

100%



  loss 0.0494413882554
  err  0.0112278005464
  (0:00:14)
Epoch 847:

100%



  loss 0.0494729988945
  err  0.0111637636612
  (0:00:16)
Epoch 848:

100%



  loss 0.0501600764317
  err  0.0111210724044
  (0:00:17)
Epoch 849:

100%



  loss 0.0510177038384
  err  0.0120389344262
  (0:00:16)
Epoch 850:

100%



  loss 0.0502636713686
  err  0.0116120218579
  (0:00:20)
Epoch 851:

100%



  loss 0.0505307607564
  err  0.0117827868852
  (0:00:13)
Epoch 852:

100%



  loss 0.0516889313728
  err  0.0115479849727
  (0:00:17)
Epoch 853:

100%



  loss 0.051324323863
  err  0.0120175887978
  (0:00:16)
Epoch 854:

100%



  loss 0.0478319472371
  err  0.0111210724044
  (0:00:13)
Epoch 855:

100%



  loss 0.0492836627817
  err  0.0121456625683
  (0:00:14)
Epoch 856:

100%



  loss 0.0501986712032
  err  0.0116974043716
  (0:00:14)
Epoch 857:

100%



  loss 0.0483142072513
  err  0.0113345286885
  (0:00:14)
Epoch 858:

100%



  loss 0.0490968543845
  err  0.0106941598361
  (0:00:17)
Epoch 859:

100%



  loss 0.0499297565122
  err  0.0114412568306
  (0:00:13)
Epoch 860:

100%



  loss 0.0520293975008
  err  0.0116120218579
  (0:00:14)
Epoch 861:

100%



  loss 0.0529431084112
  err  0.0124231557377
  (0:00:17)
Epoch 862:

100%



  loss 0.0514529555822
  err  0.0124658469945
  (0:00:15)
Epoch 863:

100%



  loss 0.051480131379
  err  0.012337773224
  (0:00:13)
Epoch 864:

100%



  loss 0.0476537931143
  err  0.0110783811475
  (0:00:14)
Epoch 865:

100%



  loss 0.0498446072216
  err  0.0121243169399
  (0:00:13)
Epoch 866:

100%



  loss 0.0505259117904
  err  0.0118895150273
  (0:00:12)
Epoch 867:

100%



  loss 0.0508793889598
  err  0.0115693306011
  (0:00:12)
Epoch 868:

100%



  loss 0.0492008585684
  err  0.0111637636612
  (0:00:12)
Epoch 869:

100%



  loss 0.0515154115614
  err  0.0127646857923
  (0:00:15)
Epoch 870:

100%



  loss 0.0503609229879
  err  0.01171875
  (0:00:13)
Epoch 871:

100%



  loss 0.0502718924557
  err  0.0118468237705
  (0:00:13)
Epoch 872:

100%



  loss 0.0492853311118
  err  0.0109929986339
  (0:00:13)
Epoch 873:

100%



  loss 0.0491151630362
  err  0.0113558743169
  (0:00:13)
Epoch 874:

100%



  loss 0.0500493553445
  err  0.0116974043716
  (0:00:12)
Epoch 875:

100%



  loss 0.0492996277906
  err  0.0113985655738
  (0:00:13)
Epoch 876:

100%



  loss 0.0510550060391
  err  0.0125512295082
  (0:00:12)
Epoch 877:

100%



  loss 0.0485666253015
  err  0.0108862704918
  (0:00:14)
Epoch 878:

100%



  loss 0.0492573699525
  err  0.0119108606557
  (0:00:14)
Epoch 879:

100%



  loss 0.0483583739839
  err  0.0112491461749
  (0:00:13)
Epoch 880:

100%



  loss 0.0504837539391
  err  0.0117614412568
  (0:00:13)
Epoch 881:

100%



  loss 0.0501850802659
  err  0.011206454918
  (0:00:13)
Epoch 882:

100%



  loss 0.0494818198304
  err  0.0114412568306
  (0:00:13)
Epoch 883:

100%



  loss 0.0504887308009
  err  0.0118254781421
  (0:00:13)
Epoch 884:

100%



  loss 0.0485909333913
  err  0.0113985655738
  (0:00:13)
Epoch 885:

100%



  loss 0.0506452381639
  err  0.0118681693989
  (0:00:13)
Epoch 886:

100%



  loss 0.0484453620627
  err  0.0107155054645
  (0:00:13)
Epoch 887:

100%



  loss 0.0513706522723
  err  0.0123804644809
  (0:00:13)
Epoch 888:

100%



  loss 0.0494423320955
  err  0.0109929986339
  (0:00:15)
Epoch 889:

100%



  loss 0.0495837164274
  err  0.0121456625683
  (0:00:13)
Epoch 890:

100%



  loss 0.048358806643
  err  0.01171875
  (0:00:15)
Epoch 891:

100%



  loss 0.052361285599
  err  0.0119962431694
  (0:00:14)
Epoch 892:

100%



  loss 0.0498165418947
  err  0.0122096994536
  (0:00:14)
Epoch 893:

100%



  loss 0.0481928075101
  err  0.0108862704918
  (0:00:15)
Epoch 894:

100%



  loss 0.0502005812287
  err  0.0126793032787
  (0:00:13)
Epoch 895:

100%



  loss 0.049636474752
  err  0.0114199112022
  (0:00:13)
Epoch 896:

100%



  loss 0.0489024829859
  err  0.011099726776
  (0:00:14)
Epoch 897:

100%



  loss 0.0504595856354
  err  0.0118895150273
  (0:00:13)
Epoch 898:

100%



  loss 0.0498579670802
  err  0.011462602459
  (0:00:14)
Epoch 899:

100%



  loss 0.0505253778073
  err  0.0121243169399
  (0:00:15)
Epoch 900:

100%



  loss 0.0497077308109
  err  0.0118041325137
  (0:00:16)
Epoch 901:

100%



  loss 0.0514039432031
  err  0.0119962431694
  (0:00:14)
Epoch 902:

100%



  loss 0.0484767814855
  err  0.0111424180328
  (0:00:14)
Epoch 903:

100%



  loss 0.0516901564405
  err  0.0118468237705
  (0:00:13)
Epoch 904:

100%



  loss 0.0506403418474
  err  0.0117614412568
  (0:00:13)
Epoch 905:

100%



  loss 0.0504239488424
  err  0.0126366120219
  (0:00:13)
Epoch 906:

100%



  loss 0.050584013521
  err  0.0124231557377
  (0:00:13)
Epoch 907:

100%



  loss 0.0484400827074
  err  0.0113131830601
  (0:00:13)
Epoch 908:

100%



  loss 0.0503055081153
  err  0.0116760587432
  (0:00:13)
Epoch 909:

100%



  loss 0.0512138071147
  err  0.0120175887978
  (0:00:13)
Epoch 910:

100%



  loss 0.0496534894944
  err  0.0115479849727
  (0:00:13)
Epoch 911:

100%



  loss 0.0477355024548
  err  0.0110356898907
  (0:00:13)
Epoch 912:

100%



  loss 0.0486710193819
  err  0.0110570355191
  (0:00:13)
Epoch 913:

100%



  loss 0.0515903902986
  err  0.011974897541
  (0:00:13)
Epoch 914:

100%



  loss 0.0496888397213
  err  0.0115479849727
  (0:00:13)
Epoch 915:

100%



  loss 0.0499282773162
  err  0.0118895150273
  (0:00:13)
Epoch 916:

100%



  loss 0.0486779912188
  err  0.0110570355191
  (0:00:13)
Epoch 917:

100%



  loss 0.0495648134355
  err  0.0115693306011
  (0:00:13)
Epoch 918:

100%



  loss 0.0506117086752
  err  0.0120175887978
  (0:00:13)
Epoch 919:

100%



  loss 0.0517506393785
  err  0.0116974043716
  (0:00:13)
Epoch 920:

100%



  loss 0.0503296177717
  err  0.011974897541
  (0:00:13)
Epoch 921:

100%



  loss 0.0506727954467
  err  0.0116760587432
  (0:00:13)
Epoch 922:

100%



  loss 0.0500783477916
  err  0.0118681693989
  (0:00:13)
Epoch 923:

100%



  loss 0.0482787937766
  err  0.0106728142077
  (0:00:13)
Epoch 924:

100%



  loss 0.0513233678076
  err  0.0118468237705
  (0:00:13)
Epoch 925:

100%



  loss 0.0508710110021
  err  0.0119322062842
  (0:00:14)
Epoch 926:

100%



  loss 0.0501912750954
  err  0.0118681693989
  (0:00:15)
Epoch 927:

100%



  loss 0.0499990018463
  err  0.011462602459
  (0:00:14)
Epoch 928:

100%



  loss 0.0496796824113
  err  0.0112918374317
  (0:00:14)
Epoch 929:

100%



  loss 0.0515048508596
  err  0.0122737363388
  (0:00:13)
Epoch 930:

100%



  loss 0.0489638029194
  err  0.011099726776
  (0:00:18)
Epoch 931:

100%



  loss 0.0506544871017
  err  0.0116974043716
  (0:00:21)
Epoch 932:

100%



  loss 0.0501916362571
  err  0.0117400956284
  (0:00:16)
Epoch 933:

100%



  loss 0.0520406895778
  err  0.012231045082
  (0:00:16)
Epoch 934:

100%



  loss 0.0492596411391
  err  0.0113772199454
  (0:00:13)
Epoch 935:

100%



  loss 0.0487378819717
  err  0.0106301229508
  (0:00:13)
Epoch 936:

100%



  loss 0.0520326556025
  err  0.0123591188525
  (0:00:14)
Epoch 937:

100%



  loss 0.0509003814452
  err  0.0118041325137
  (0:00:13)
Epoch 938:

100%



  loss 0.0528931171364
  err  0.0122523907104
  (0:00:13)
Epoch 939:

100%



  loss 0.0529561976494
  err  0.0120175887978
  (0:00:13)
Epoch 940:

100%



  loss 0.0488930415512
  err  0.0116547131148
  (0:00:13)
Epoch 941:

100%



  loss 0.0506149959105
  err  0.0115479849727
  (0:00:14)
Epoch 942:

100%



  loss 0.0479803629197
  err  0.0110356898907
  (0:00:14)
Epoch 943:

100%



  loss 0.0501608879667
  err  0.01171875
  (0:00:13)
Epoch 944:

100%



  loss 0.0479011872034
  err  0.0111210724044
  (0:00:15)
Epoch 945:

100%



  loss 0.0526663968244
  err  0.0122950819672
  (0:00:10)
Epoch 946:

100%



  loss 0.0498982068988
  err  0.0116333674863
  (0:00:10)
Epoch 947:

100%



  loss 0.0504646545004
  err  0.0121883538251
  (0:00:10)
Epoch 948:

100%



  loss 0.0492840541846
  err  0.0113345286885
  (0:00:10)
Epoch 949:

100%



  loss 0.0514308953597
  err  0.0116333674863
  (0:00:10)
Epoch 950:

100%



  loss 0.0509538842127
  err  0.0118254781421
  (0:00:10)
Epoch 951:

100%



  loss 0.0510072157568
  err  0.01171875
  (0:00:09)
Epoch 952:

100%



  loss 0.0507144566817
  err  0.0116120218579
  (0:00:10)
Epoch 953:

100%



  loss 0.0501742083388
  err  0.0115266393443
  (0:00:11)
Epoch 954:

100%



  loss 0.049322216082
  err  0.0115693306011
  (0:00:13)
Epoch 955:

100%



  loss 0.0512824971502
  err  0.0128287226776
  (0:00:11)
Epoch 956:

100%



  loss 0.0506181859531
  err  0.0117614412568
  (0:00:11)
Epoch 957:

100%



  loss 0.0467624228519
  err  0.0106941598361
  (0:00:11)
Epoch 958:

100%



  loss 0.0512477770547
  err  0.0121670081967
  (0:00:10)
Epoch 959:

100%



  loss 0.0510073282439
  err  0.0116333674863
  (0:00:10)
Epoch 960:

100%



  loss 0.0489928407882
  err  0.0113131830601
  (0:00:10)
Epoch 961:

100%



  loss 0.0496104461239
  err  0.0117827868852
  (0:00:10)
Epoch 962:

100%



  loss 0.0494608834654
  err  0.011462602459
  (0:00:11)
Epoch 963:

100%



  loss 0.0504253576153
  err  0.0119322062842
  (0:00:10)
Epoch 964:

100%



  loss 0.0507117074239
  err  0.0118681693989
  (0:00:11)
Epoch 965:

100%



  loss 0.0507958654188
  err  0.0117400956284
  (0:00:11)
Epoch 966:

100%



  loss 0.0498854559846
  err  0.0122096994536
  (0:00:12)
Epoch 967:

100%



  loss 0.0515350369257
  err  0.0121456625683
  (0:00:12)
Epoch 968:

100%



  loss 0.0522950488121
  err  0.012593920765
  (0:00:10)
Epoch 969:

100%



  loss 0.0494621396745
  err  0.0116974043716
  (0:00:10)
Epoch 970:

100%



  loss 0.0499013227759
  err  0.0110356898907
  (0:00:11)
Epoch 971:

100%



  loss 0.0488682120626
  err  0.0115479849727
  (0:00:10)
Epoch 972:

100%



  loss 0.0499610131081
  err  0.0115052937158
  (0:00:10)
Epoch 973:

100%



  loss 0.0494877424993
  err  0.0110356898907
  (0:00:10)
Epoch 974:

100%



  loss 0.0506632345356
  err  0.0119322062842
  (0:00:10)
Epoch 975:

100%



  loss 0.0509180836609
  err  0.011974897541
  (0:00:11)
Epoch 976:

100%



  loss 0.0505795557093
  err  0.0123164275956
  (0:00:11)
Epoch 977:

100%



  loss 0.0484453404075
  err  0.0114412568306
  (0:00:10)
Epoch 978:

100%



  loss 0.0501502828178
  err  0.0113345286885
  (0:00:10)
Epoch 979:

100%



  loss 0.0507966169987
  err  0.0124231557377
  (0:00:10)
Epoch 980:

100%



  loss 0.0491274274412
  err  0.0115906762295
  (0:00:10)
Epoch 981:

100%



  loss 0.0513469761319
  err  0.012337773224
  (0:00:10)
Epoch 982:

100%



  loss 0.0494074126797
  err  0.0117614412568
  (0:00:10)
Epoch 983:

100%



  loss 0.050446823976
  err  0.0112491461749
  (0:00:10)
Epoch 984:

100%



  loss 0.0514749907189
  err  0.012337773224
  (0:00:10)
Epoch 985:

100%



  loss 0.0526031377373
  err  0.0132342896175
  (0:00:10)
Epoch 986:

100%



  loss 0.0501711624038
  err  0.0116333674863
  (0:00:10)
Epoch 987:

100%



  loss 0.0509625573762
  err  0.0114199112022
  (0:00:10)
Epoch 988:

100%



  loss 0.0504815208502
  err  0.012231045082
  (0:00:10)
Epoch 989:

100%



  loss 0.0519790488456
  err  0.012231045082
  (0:00:10)
Epoch 990:

100%



  loss 0.046804674125
  err  0.0104380122951
  (0:00:11)
Epoch 991:

100%



  loss 0.0476325072381
  err  0.0106941598361
  (0:00:11)
Epoch 992:

100%



  loss 0.0478471450214
  err  0.0113558743169
  (0:00:10)
Epoch 993:

100%



  loss 0.0505198857727
  err  0.0121243169399
  (0:00:10)
Epoch 994:

100%



  loss 0.0500504379466
  err  0.010950307377
  (0:00:11)
Epoch 995:

100%



  loss 0.0508021946979
  err  0.0117827868852
  (0:00:10)
Epoch 996:

100%



  loss 0.0501218303826
  err  0.0118041325137
  (0:00:10)
Epoch 997:

100%



  loss 0.0519130192175
  err  0.0117827868852
  (0:00:10)
Epoch 998:

100%



  loss 0.0503042559248
  err  0.0118468237705
  (0:00:10)
Epoch 999:

100%



  loss 0.0506622662114
  err  0.0120175887978
  (0:00:10)
Epoch 1000:
  loss 0.0506539636977
  err  0.0117827868852
  (0:00:10)
accuracy score : 0.97354978355
classification report : 
             precision    recall  f1-score   support

          0       0.98      0.98      0.98      2278
          1       0.98      0.99      0.98      2574
          2       0.97      0.97      0.97      2334
          3       0.97      0.96      0.97      2387
          4       0.97      0.97      0.97      2290
          5       0.97      0.97      0.97      2036
          6       0.98      0.99      0.98      2266
          7       0.98      0.96      0.97      2403
          8       0.97      0.96      0.97      2232
          9       0.96      0.97      0.96      2300

avg / total       0.97      0.97      0.97     23100

confusion matrix : 
[[2241    1    5    2    1    6    8    5    6    3]
 [   0 2539   11    8    2    0    5    6    3    0]
 [  11    7 2265    8    7    3    8   14    6    

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.3,
    learn_rate_decays = 0.9,
    epochs = 10,
    verbose = 1)
accuracy score : 0.977142857143

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.3,
    learn_rate_decays = 0.9,
    epochs = 20,
    verbose = 1)
accuracy score : 0.980779220779

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.3,
    learn_rate_decays = 0.9,
    epochs = 30,
    verbose = 1)
accuracy score : 0.981082251082

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.4,
    learn_rate_decays = 0.9,
    epochs = 20,
    verbose = 1)
accuracy score : 0.979047619048

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.2,
    learn_rate_decays = 0.9,
    epochs = 20,
    verbose = 1)
accuracy score : 0.981038961039

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.5,
    learn_rate_decays = 0.9,
    epochs = 20,
    verbose = 1)
accuracy score : 0.971601731602

dbn = DBN(
    [trainX.shape[1], 400, 10],
    learn_rates = 0.3,
    learn_rate_decays = 0.9,
    epochs = 20,
    verbose = 1)
accuracy score : 0.981948051948

dbn = DBN(
    [trainX.shape[1], 500, 10],
    learn_rates = 0.3,
    learn_rate_decays = 0.9,
    epochs = 20,
    verbose = 1)
accuracy score : 0.981255411255

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.001,
    learn_rate_decays = 0.9,
    epochs = 20,
    verbose = 1)
accuracy score : 0.923722943723


dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.01,
    learn_rate_decays = 0.9,
    epochs = 100,
    verbose = 1)
accuracy score : 0.973246753247

dbn = DBN(
    [trainX.shape[1], 300, 10],
    learn_rates = 0.01,
    learn_rate_decays = 0.9,
    epochs = 1000,
    verbose = 1)
accuracy score : 0.97354978355